<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Banana - Processing of 16S rRNA reads obtained from Illumina MiSeq platform</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Banana</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html" rel="" target="">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-processing-of-miseq-reads" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Processing of MiSeq reads</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-processing-of-miseq-reads">    
        <li>
    <a class="dropdown-item" href="./Bacteria_reads_processing.html" rel="" target="">
 <span class="dropdown-text">16S rRNA reads</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_reads_processing.html" rel="" target="">
 <span class="dropdown-text">ITS2 reads</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-ecological-analyses" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Ecological analyses</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-ecological-analyses">    
        <li>
    <a class="dropdown-item" href="./Bacteria_data_analysis.html" rel="" target="">
 <span class="dropdown-text">Bacterial dataset</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_data_analysis.html" rel="" target="">
 <span class="dropdown-text">Fungal dataset</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/anitalasa" rel="" target=""><i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://orcid.org/0000-0003-3783-7157" rel="" target="">
 <span class="menu-text"><img src="images/orcid.png" title="ORCID" alt="ORCID" style="height:20px;"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://grupos.eez.csic.es/mae/" rel="" target="">
 <span class="menu-text"><img src="images/internet.png" title="web research group" alt="web research group" style="height:20px;"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#formatting-the-name-of-the-samples" id="toc-formatting-the-name-of-the-samples" class="nav-link active" data-scroll-target="#formatting-the-name-of-the-samples">1. Formatting the name of the samples</a></li>
  <li><a href="#check-the-quality-of-the-sequencing" id="toc-check-the-quality-of-the-sequencing" class="nav-link" data-scroll-target="#check-the-quality-of-the-sequencing">2. Check the quality of the sequencing</a>
  <ul class="collapse">
  <li><a href="#a-count-the-number-of-reads" id="toc-a-count-the-number-of-reads" class="nav-link" data-scroll-target="#a-count-the-number-of-reads">a) Count the number of reads</a></li>
  <li><a href="#b-inspect-the-length-of-the-reads" id="toc-b-inspect-the-length-of-the-reads" class="nav-link" data-scroll-target="#b-inspect-the-length-of-the-reads">b) Inspect the length of the reads</a></li>
  <li><a href="#c-check-the-quality-plots" id="toc-c-check-the-quality-plots" class="nav-link" data-scroll-target="#c-check-the-quality-plots">c) Check the quality plots</a></li>
  </ul></li>
  <li><a href="#figaro-tool" id="toc-figaro-tool" class="nav-link" data-scroll-target="#figaro-tool">3. FIGARO tool</a></li>
  <li><a href="#filter-and-trimming-step" id="toc-filter-and-trimming-step" class="nav-link" data-scroll-target="#filter-and-trimming-step">4. Filter and trimming step</a></li>
  <li><a href="#cutadapt-removal-of-the-primers" id="toc-cutadapt-removal-of-the-primers" class="nav-link" data-scroll-target="#cutadapt-removal-of-the-primers">5. Cutadapt: removal of the primers</a></li>
  <li><a href="#dada2-machine-learning" id="toc-dada2-machine-learning" class="nav-link" data-scroll-target="#dada2-machine-learning">6. DADA2: machine learning</a></li>
  <li><a href="#dada2-sample-inference" id="toc-dada2-sample-inference" class="nav-link" data-scroll-target="#dada2-sample-inference">7. DADA2: Sample inference</a></li>
  <li><a href="#overlap-f-and-r-sequences" id="toc-overlap-f-and-r-sequences" class="nav-link" data-scroll-target="#overlap-f-and-r-sequences">8. Overlap F and R sequences</a></li>
  <li><a href="#merge-sequence-tables-from-different-runs" id="toc-merge-sequence-tables-from-different-runs" class="nav-link" data-scroll-target="#merge-sequence-tables-from-different-runs">9. Merge sequence tables from different runs</a></li>
  <li><a href="#chimeral-removal" id="toc-chimeral-removal" class="nav-link" data-scroll-target="#chimeral-removal">10. Chimeral removal</a></li>
  <li><a href="#other-steps-flter-the-asvs-by-their-length" id="toc-other-steps-flter-the-asvs-by-their-length" class="nav-link" data-scroll-target="#other-steps-flter-the-asvs-by-their-length">11. Other steps: flter the ASVs by their length</a></li>
  <li><a href="#taxonomic-classification" id="toc-taxonomic-classification" class="nav-link" data-scroll-target="#taxonomic-classification">12. Taxonomic classification</a></li>
  <li><a href="#mock-community-setting-the-sequencing-detection-limit" id="toc-mock-community-setting-the-sequencing-detection-limit" class="nav-link" data-scroll-target="#mock-community-setting-the-sequencing-detection-limit">13. MOCK Community: setting the sequencing detection limit</a></li>
  <li><a href="#removal-of-erronous-sequences-and-taxonomy-refinement" id="toc-removal-of-erronous-sequences-and-taxonomy-refinement" class="nav-link" data-scroll-target="#removal-of-erronous-sequences-and-taxonomy-refinement">14. Removal of erronous sequences and taxonomy refinement</a></li>
  <li><a href="#extra-code" id="toc-extra-code" class="nav-link" data-scroll-target="#extra-code">15. Extra code</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Processing of <em>16S rRNA</em> reads obtained from Illumina MiSeq platform</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>First of all, we will install and load of the required packages and libraries</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"nuriamw/micro4all"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(micro4all)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>As you can see, we are going to use different functions of <a href="https://benjjneb.github.io/dada2/tutorial.html" title="Click to DADA2 website">DADA2 package</a>, so it will be nice if you read all the documentation regarding this package.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>The steps 1 to 8 must be performed for each sequencing run. If you are working with different sequencing runs, you have to join all the sequence tables into one object (step 9). Once the joined element is obtained, steps 10-last step have to be applied to the joined object.</p>
</div>
</div>
<section id="formatting-the-name-of-the-samples" class="level1">
<h1>1. Formatting the name of the samples</h1>
<p>Now, we will start! Set the working directory and specify the path where you are working:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>path<span class="ot">=</span> <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads"</span><span class="co">#insert here the path where your fastq files are</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path) <span class="co">#Check that all files are here included (script + fastq files)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Sort F and R reads separately and save them into two variables:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#our filenames have the format "NGS015-23-16S-A2S12R_S51_L001_R2_001.fastq.gz"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Extract the name (code) of your samples, and remove all the extra information from the filenames:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">4</span>) </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#split the string when "-" is found and keep the 4th part of the string</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we will get, for example, this: "A2S12R_S51_L001_R2_001.fastq.gz""</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(sample.names_raw), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co">#removal of extra information (in this example, split sample.names_raw when "_" is found, and get the first part of the string)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We are going to rename the samples corresponding to the mock community, so that they satisfy the requirement of the functions to be applied in next steps:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK1"</span>, <span class="st">"MOCK-1"</span>, sample.names)<span class="co">#replace "MOCK1" by "MOCK-1"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK2"</span>, <span class="st">"MOCK-2"</span>, sample.names)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK3"</span>, <span class="st">"MOCK-3"</span>, sample.names)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>sample.names <span class="co">#check that the name of our samples is OK</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="check-the-quality-of-the-sequencing" class="level1">
<h1>2. Check the quality of the sequencing</h1>
<p>There are different ways to check the quality of the reads:</p>
<section id="a-count-the-number-of-reads" class="level2">
<h2 class="anchored" data-anchor-id="a-count-the-number-of-reads">a) Count the number of reads</h2>
<p>It would be nice to check whether we obtained enough reads from the sequencing service. Otherwise, we should ask the service to repeat the sequencing.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)){</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>    raw_reads_count <span class="ot">=</span> <span class="fu">rbind</span>(raw_reads_count, </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  } <span class="co">#this loop counts the number of F reads by means of the ShortRead package</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count)<span class="ot">=</span> sample.names <span class="co">#formatting of the output</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>a<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>raw_reads_count2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnRs)){<span class="co">#do the same with R reads</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>  raw_reads_count2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count2, </span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnRs[i])))</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count2)<span class="ot">=</span> sample.names</span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count2)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>b<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count2),raw_reads_count2)</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>a<span class="sc">==</span>b <span class="co">#check that we get the same number of F and R reads</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)],</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">min</span>(raw_reads_count))<span class="co">#check which sample accounts for the lowest number of reads. Be careful if you keep the negative control of the sequencing</span></span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)],</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">max</span>(raw_reads_count))<span class="co">#check which sample accounts for the highest number of reads.</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count),</span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">file=</span><span class="st">"Num_raw_reads_16s.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(raw_reads_count) <span class="co">#please, check the number of reads per sample</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>At this point, we have to check the number of raw reads per sample. The researcher has to verify whether this number is enough or not. It depends on the type of sample, among other parameters.</p>
</section>
<section id="b-inspect-the-length-of-the-reads" class="level2">
<h2 class="anchored" data-anchor-id="b-inspect-the-length-of-the-reads">b) Inspect the length of the reads</h2>
<p>In our case, the genomics service followed 2x275 bp and 2x300 bp PE strategy, so most of the sequences are expected to have 275 or 300 bp.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>reads<span class="ot">=</span>ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co">#save the reads in a new variable with the format of ShortRead package, which is a bit different from standard R variable</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">=</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co">#get the length of the reads</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>counts<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  counts<span class="ot">=</span><span class="fu">rbind</span>(counts,</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>               <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width<span class="sc">==</span>uniques[i])))</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>}<span class="co">#loop to count the number of reads of each length.  </span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">=</span>  <span class="fu">cbind</span>(uniques,counts)</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="co">#check the histogram</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[,<span class="dv">1</span>],<span class="at">decreasing =</span> <span class="cn">TRUE</span>),]) <span class="co">#Most of the sequences should fall in expected sequence length</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting</span></span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main=</span><span class="st">"Forward length distribution"</span>,</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Sequence length"</span>, <span class="at">ylab=</span><span class="st">"Raw reads"</span>)</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file=</span><span class="st">"Lenght_raw_reads_16S.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="c-check-the-quality-plots" class="level2">
<h2 class="anchored" data-anchor-id="c-check-the-quality-plots">c) Check the quality plots</h2>
<p>In this step, keep in mind the expected length of your amplicon. Then, check if the overlapping of F and R reads is possible considering the quality of the last nucleotides.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])<span class="co">#select the specific samples you want to check, in this case 4 and 5</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="figaro-tool" class="level1">
<h1>3. FIGARO tool</h1>
<p>We will use FIGARO to determine the best position of trimming in both F and R reads, and the best maximum expected errors for DADA2. <strong>WARNING</strong>: FIGARO does not run in R (not even in Windows!), so we will run it in python3</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>figFs <span class="ot">=</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnFs)) <span class="co">#create the directory where the output will be saved</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>figRs <span class="ot">=</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnRs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>FIGARO does not work if all the samples are of different lengths. Thus, we have to cut the sequences so that this tool works. Do not panic because this cut is just made in this step. Then, we will work with the full-length reads</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>out.figaro<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, figFs, fnRs, figRs,<span class="at">compress=</span><span class="cn">TRUE</span>, </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">271</span>,<span class="dv">271</span>)) </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co">#select the position so that most of the sequences are considered </span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">#(check the histogram or the dataframe with the lengths of the reads to select this "pseudo"trimming positions)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>figaro<span class="ot">=</span><span class="fu">system</span>((<span class="st">"python3 /home/programas/figaro/figaro/figaro.py  -i ~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/figaro -o ~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/figaro -a 426 -f 17 -r 21"</span>), <span class="at">intern=</span><span class="cn">TRUE</span>) </span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#indicate the path where the script "figaro.py" is located</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#-a, indicate the length of the amplicon, </span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#f, indicate the length of primer F</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="co">#r, indicate the length R  primer</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(figaro) <span class="co">#select the best parameters proposed by FIGARO, here: Trimming position F,R: 248,236; maximum expected error F, R: 3,3.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filter-and-trimming-step" class="level1">
<h1>4. Filter and trimming step</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>filtFs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))<span class="co">#create the "filtered" directory</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>filtRs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>out<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs, <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">248</span>,<span class="dv">236</span>),</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>), <span class="at">truncQ=</span><span class="dv">2</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress=</span><span class="cn">TRUE</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">minLen=</span><span class="dv">50</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#truncLen: introduce the best trimming position proposed by FIGARO</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#maxN=0 : remove the reads with at least 1 ambiguity</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co">#trunQ=2: remove all the reads with one nucleotide with 2 or less quality value</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co">#maxEE: maximum expected error for F and R reads proposed by FIGARO</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#minLen=50: remove reads shorter than 50 bp</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Our reads are partially filtered and trimmed (still have the primers!)</p>
<p>Here you have the parameters we obtained for each sequencing runs:</p>
<table class="table">
<thead>
<tr class="header">
<th>Run</th>
<th>Positions</th>
<th></th>
<th>maxEE</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td><strong>F</strong></td>
<td><strong>R</strong></td>
<td><strong>F</strong></td>
<td><strong>R</strong></td>
</tr>
<tr class="even">
<td><strong>Run1</strong></td>
<td>248</td>
<td>236</td>
<td>3</td>
<td>3</td>
</tr>
<tr class="odd">
<td><strong>Run2</strong></td>
<td>272</td>
<td>212</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="even">
<td><strong>Run3</strong></td>
<td>278</td>
<td>206</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td><strong>Run4</strong></td>
<td>273</td>
<td>211</td>
<td>2</td>
<td>2</td>
</tr>
</tbody>
</table>
</section>
<section id="cutadapt-removal-of-the-primers" class="level1">
<h1>5. Cutadapt: removal of the primers</h1>
<p>Our reads still have the primers (which are artificial sequences and can have ambiguities, <em>N</em> nucleotides). So, we have to remove them and discard all the sequences in which primers are not found (because if the sequencing run was ok, primers should had been found inside the reads). Letâ€™s use cutadapt tool, which does not run in R but in python:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>FWD<span class="ot">=</span><span class="fu">c</span>(<span class="st">"CCTACGGGNBGCASCAG"</span>) <span class="co">#insert the sequences of the primers</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>REV<span class="ot">=</span><span class="fu">c</span>(<span class="st">"GACTACNVGGGTATCTAATCC"</span>) <span class="co">#here we have 2 degenerations (ambiguities)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">=</span> <span class="cf">function</span>(primer) {</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">=</span><span class="fu">DNAString</span>(primer)  <span class="co">#package Biostrings does not work with characters but with strings, so let's change the class of the primers variables.</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">=</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna), <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co">#change from string to character</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>} <span class="co">#this function calculates all the possible orientations of the primers</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">=</span> <span class="fu">allOrients</span>(FWD) <span class="co">#pass the function to F and R primers</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>REV.orients  <span class="ot">=</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>REV.orients</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">=</span><span class="cf">function</span>(primer, fn) {</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">=</span><span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>} <span class="co">#it counts the number of sequence in each orientation</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),<span class="co">#check for instante, in sample number 4</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]))</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="co">#here we make a summary table. Please, visit DADA2 website for a better interpretation of the table.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now yes, we are running cutadapt, although we have to adapt the reads previously for cutadapt. This tool needs some flags that have to be added to the reads. Visit the web of <a href="https://cutadapt.readthedocs.io/en/stable/guide.html" title="User guide of cutadapt">cutadapt</a> for more information.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>cutadapt  <span class="ot">=</span>  <span class="st">"/usr/bin/cutadapt"</span> <span class="co">#path to cutadapt </span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>) <span class="co">#create a directory where processed reads will be saved</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce arguments for cutadapt</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)  </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>R1.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>,FWD,<span class="st">"..."</span>, REV.RC) <span class="co">#adding the flags</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>R2.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-A"</span>,<span class="st">" "</span>,<span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="co">#run cutadapt</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>,<span class="st">"-m"</span>, <span class="dv">1</span>, <span class="st">"--discard-untrimmed"</span>, <span class="st">"-j"</span>,<span class="dv">0</span>, <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], filtFs[i], filtRs[i],<span class="st">"--report=minimal"</span>)) </span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a><span class="co">#-n 2: remove the primers</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="co">#-m 1: remove empty sequences</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="co">#-j 0: automatically detect number of cores</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="co">#-0: output files</span></span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a><span class="co">#-i: input files. We use the filtered and trimmed reads</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we will check whether cutadapt has removed all the primers. For that purpose, we pass the previously created function to check the number of times each primer appears in our dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),<span class="co">#Here we can indicate the number of the sample we want to check. In this case, sample number 4</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note that in some cases, after the removal of the primers by cutadapt, the function <em>primerHits</em> still finds some primers in our dataset. Do no panic, these differences are normal because cutadapt and DADA2 functions are based on different algorithms.</p>
</div>
</div>
</section>
<section id="dada2-machine-learning" class="level1">
<h1>6. DADA2: machine learning</h1>
<p>We are going to use DADA2 to denoise, infer the samples and to correct the expected errors by applying different functions of the package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First, learn error rates</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">=</span><span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">=</span><span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets plot the errors in each position</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co">#these plots show the possible errors made for each possible mutation (A&gt;C, A&gt;G, etc). Dots should be close to the red line.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dada2-sample-inference" class="level1">
<h1>7. DADA2: Sample inference</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">=</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err=</span>errF, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">=</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err=</span>errR, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">4</span>]]<span class="co">#check the inferred fourth sample (it's just an example). We will obtain the number of true sequence variants from the X number of unique sequences.</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">=</span> sample.names</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">=</span> sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="overlap-f-and-r-sequences" class="level1">
<h1>8. Overlap F and R sequences</h1>
<p>In this step we are going to overlap F (forward) and R (reverse) reads coming from the same sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>mergers<span class="ot">=</span><span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">4</span>]]) <span class="co">#check the output for sample number 4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we are going to construct an amplicon sequence variant table (ASV table).</p>
<p>Be careful, because we will get one ASV table per sequencing run. Thus, as we got 4 runs for bacterial amplicons, we will get 4 different ASV tables.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>seqtab_run1<span class="ot">=</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_run1) <span class="co">#indicates the number of samples (including mock community and negative controls) and the number of ASVs</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_run1, <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/seqtab_run1.rds"</span>) <span class="co">#save the seqtab in .rds format</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merge-sequence-tables-from-different-runs" class="level1">
<h1>9. Merge sequence tables from different runs</h1>
<p>As indicated before, in this step we are going to merge the sequence tables (seqtabs) coming from different runs into just one table. So, firstly, we have to load all of them to the current project. Please, note that all the seqtabs have to be in the same working directory!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>seqtab_run1<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run1.rds"</span>)<span class="co">#load all the .rds files in the same directory</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>seqtab_run2<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run2.rds"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>seqtab_run3<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run3.rds"</span>)</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>seqtab_run4<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run4.rds"</span>)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>mergedSeqTab<span class="ot">=</span><span class="fu">mergeSequenceTables</span>(seqtab_run1,seqtab_run2,seqtab_run3,seqtab_run4,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">repeats=</span><span class="st">"sum"</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="co">#with "sum" we indicate that we want to join and sum all the reads that are in samples with the same name. Why do we do that? Because in each sequencing run we have 3 replicates of the mock community, and we are going to sum the same replicates from different runs.</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co">#e.g., MOCK-1 (run1) + MOCK-1 (run2)+ MOCK-1 (run3)+ MOCK-1 (run4)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="chimeral-removal" class="level1">
<h1>10. Chimeral removal</h1>
<p>As the sequencing process is based on PCRs, chimeric sequences are expect to appear. So, letâ€™s remove them.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method=</span><span class="st">"consensus"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co">#indicates the number of samples and ASVs (but not the number of sequences)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="other-steps-flter-the-asvs-by-their-length" class="level1">
<h1>11. Other steps: flter the ASVs by their length</h1>
<p>Letâ€™s check the number of ASVs and their length</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))  <span class="co">#Number of ASV of each length</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#first line: length of the sequences</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#second line: number of sequences of each length</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Our amplicon is of 426 bp in length, so most of the sequences would have that length</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate the number of sequences of each ASV that of each corresponding length:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">=</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">length=</span><span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count=</span>reads.per.seqlen)</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>count)) <span class="sc">+</span> <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we have to filter the length of the ASVs. For that purpose, we have to keep in mind the data from the dataframe and the histogram. Choose those lengths in which we can retrieve enough sequences (&gt;10000 sequences, if possible)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> seqtab.nochim[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab.nochim)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">401</span>,<span class="dv">428</span>)] <span class="co">#the consensus sequences of our ASVs will be from 401 to 428 nucleotides</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxonomic-classification" class="level1">
<h1>12. Taxonomic classification</h1>
<p>We are going to use the database of the Ribosomal Database Project (RDP), specifically the trainset 19.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>taxa_rdp2 <span class="ot">=</span><span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/home/databases/rdp_train_set_19_H.fa"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co">#we selected the last version of the RDP-II training set v.19</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">=</span> seqtab.nochim <span class="co">#edit the tables</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">=</span> <span class="fu">t</span>(ASV)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na <span class="ot">=</span> <span class="fu">apply</span>(taxa_rdp2,<span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[,<span class="sc">-</span><span class="dv">7</span>] <span class="co">#replace the "NA" values by "unclassified" in those case in which an ASV have not been identified at a specific taxonomic rank</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na<span class="ot">=</span>taxa_rdp_na[,<span class="sc">-</span>(<span class="dv">7</span><span class="sc">:</span><span class="dv">15</span>)]</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a><span class="co">#we are going to modify a bit the tables so that the name of ASVs looks like "ASV00001" (in case we have 10000 ASVs or more). In case we have just 100 ASVs, they will have that code: "ASV001"</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">=</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>names <span class="ot">=</span><span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co">#As many 0 as digits</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>ASV_names<span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_rdp_na,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASVt,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">=</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_table_classified_raw, <span class="at">file=</span><span class="st">"ASV_table_classified_raw_wirhMockwithPlastids.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we have our ASV table with the corresponding taxonomic classification. But, it still includes mitochondria and plastids from the host plants, the mock community and other invalid sequences.</p>
</section>
<section id="mock-community-setting-the-sequencing-detection-limit" class="level1">
<h1>13. MOCK Community: setting the sequencing detection limit</h1>
<p>We included a mock community in order to establish the detection limit of the sequencing process. Now, we have to address this cut-off. For that purpose, we will run the function <em>MockCommunity</em> which will ask us whether specific microorganisms are included in the mock community. We have to answer accordingly, and the function will go on asking us until an ASV not included in the mock community is found. Its relative abundance in the whole dataset will be considered as the sequencing detection limit, since it should have not been detected if it was not included in the mock community (a false detection, so all the ASV whose relative abundance is below this cut-off, are not real ASVs). Then, we have to remove the mock community.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>ASV_filtered_MOCK<span class="ot">=</span><span class="fu">MockCommunity</span>(ASV_table_classified_raw,mock_composition,<span class="at">ASV_column =</span> <span class="st">"ASV_names"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
</div>
<div class="callout-body-container callout-body">
<p>Be careful with ASVs belonging to genus <em>Limosilactobacillus</em> (especially if their relative abundance is high). Possibly, they are missclassified and belong to genus <em>Bacillus</em>, which is a member of the MockCommunity here employed as well.</p>
</div>
</div>
</section>
<section id="removal-of-erronous-sequences-and-taxonomy-refinement" class="level1">
<h1>14. Removal of erronous sequences and taxonomy refinement</h1>
<p>This step is specially important if we are sequencing plant tissues (root, phyllosphere, etc), because with the primers employed here it is possible to amplify plant hostsâ€™ DNA, for example, that corresponding to mitochondria, chloroplasts, among others. Thus, letâ€™s remove all these undesired sequences. We will also remove those ASVs not classified even at kingdom level:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>ASV_final<span class="ot">=</span>ASV_filtered_MOCK[(<span class="fu">which</span>(ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Streptophyta"</span>  <span class="sc">&amp;</span>ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Mitochondria"</span> <span class="sc">&amp;</span>  ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Chlorophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Streptophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Bacillariophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Mitochondria"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Class<span class="sc">!=</span><span class="st">"Chloroplast"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Order<span class="sc">!=</span><span class="st">"Chloroplast"</span> <span class="sc">&amp;</span>ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Chloroplast"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"unclassified"</span>)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Cyanobacterial sequences could be tricky, since they could be really close to plant sequences (chloroplasts). Thus, we should retrieve those sequences ascribed to the phylym <em>Cyanobacteriota</em>, and we then remove those that are not classified at class level. In our case, all the Cyanobacteria were chloroplasts (Phylum <em>Cyanobacteriota</em>, class Chloroplast), hence, they were removed in the previous step. But we encourage to make a BLAST at this point.</p>
<p>We are going to check also the ASVs not classified at Phylum level</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>unclass_phy <span class="ot">=</span> ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"unclassified"</span>),] <span class="co">#save them into a new variable</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>seq_unclass <span class="ot">=</span> <span class="fu">as.list</span>(unclass_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclass, <span class="at">names=</span>unclass_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"unclassified_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>) <span class="co">#write the sequences in fasta format</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we align by BLASTn these unclassified sequences against those held in the NCBI nt database, which is already downloaded in our PC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query unclassified_phylum.fas -db /home/databases/nt -out unclassified_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 5 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we checked manually the output of BLAST and removed one ASV (ASV01487, unclassified at Phylum level with the RDP-II database) which was classified as *Musa* sp.</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters<span class="ot">=</span>ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV01487"</span>)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here we removed the sample corresponding to the negative control of the sequencing run (named â€œCnegâ€).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters<span class="ot">=</span><span class="fu">subset</span>(ASV_final_all_filters, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(Cneg))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>And finally, letâ€™s save the definitive bacterial ASV table:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">" "</span><span class="ot">=</span><span class="fu">rownames</span>(ASV_final_all_filters),ASV_final_all_filters),<span class="at">file=</span><span class="st">"ASV_Bacterias_final.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will also save the fasta file and perform a phylogenetic tree with <strong>all the ASVs</strong> because we will need a phylogenetic tree to perform further ecological analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>seq_fasta <span class="ot">=</span> <span class="fu">as.list</span>(ASV_final_all_filters<span class="sc">$</span>ASV_seqs) </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_fasta, <span class="at">names=</span>ASV_final_all_filters<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"fasta_para_arbol16S.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extra-code" class="level1">
<h1>15. Extra code</h1>
<p>As stated before, we will need a phylogenetic tree to calculate Weighted UniFrac distances among samples. Thus, here you have the code needed to calculate the tree (supposing you will calculate it in the same working path in which you have saved the fasta file <em>seq_fasta</em>)</p>
<p>Firstly, we have to align all the sequences. For that purpose, we will use <a href="https://mafft.cbrc.jp/alignment/server/index.html" title="Click to MAFFT website">MAFFT software</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>mafft <span class="ot">=</span> <span class="st">"/usr/bin/mafft"</span> <span class="co">#set MAFFT's path</span></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(mafft, <span class="at">args=</span><span class="fu">c</span>(<span class="st">"--auto"</span>, <span class="st">"fasta_para_arbol16S.fas&gt;"</span>, <span class="st">"alignment"</span>))<span class="co">#after "--auto", set the name of the fasta file, followed by the name of the alignment.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we are going to calculate the phylogenetic tree with <a href="http://www.microbesonline.org/fasttree/" title="Click to FastTree website">FastTree MP tool</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>FastTreeMP<span class="ot">=</span> <span class="st">"/home/programas/FastTreeMP/FastTreeMP"</span><span class="co">#set the path of the software</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(FastTreeMP, <span class="at">args=</span><span class="st">"--version"</span> )</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(FastTreeMP, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"-gamma"</span>, <span class="st">"-nt"</span>, <span class="st">"-gtr"</span>, <span class="st">"-spr"</span>,<span class="dv">4</span> ,<span class="st">"-mlacc"</span>, <span class="dv">2</span>, <span class="st">"-slownni"</span>, <span class="st">"&lt;alignment&gt;"</span>, <span class="st">"tree"</span>))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Our tree is Gamma20-based likelihood tree, based on the input alignment named "alignment", and the output tree is named "tree".</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Click <a href="./Bacteria_data_analysis.html" title="Bacterial data analyses">here</a> to visit the scripts needed for the ecological analyses.</p>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb38" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Processing of *16S rRNA* reads obtained from Illumina MiSeq platform"</span></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>First of all, we will install and load of the required packages and libraries</span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"nuriamw/micro4all"</span>)</span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(micro4all)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>As you can see, we are going to use different functions of <span class="co">[</span><span class="ot">DADA2 package</span><span class="co">](https://benjjneb.github.io/dada2/tutorial.html "Click to DADA2 website")</span>, so it will be nice if you read all the documentation regarding this package.</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>::: callout-important</span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>The steps 1 to 8 must be performed for each sequencing run. If you are working with different sequencing runs, you have to join all the sequence tables into one object (step 9). Once the joined element is obtained, steps 10-last step have to be applied to the joined object.</span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1. Formatting the name of the samples</span></span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>Now, we will start! Set the working directory and specify the path where you are working:</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a>path<span class="ot">=</span> <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads"</span><span class="co">#insert here the path where your fastq files are</span></span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path) <span class="co">#Check that all files are here included (script + fastq files)</span></span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>Sort F and R reads separately and save them into two variables:</span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-56"><a href="#cb38-56" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-57"><a href="#cb38-57" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb38-58"><a href="#cb38-58" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb38-59"><a href="#cb38-59" aria-hidden="true" tabindex="-1"></a><span class="co">#our filenames have the format "NGS015-23-16S-A2S12R_S51_L001_R2_001.fastq.gz"</span></span>
<span id="cb38-60"><a href="#cb38-60" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-61"><a href="#cb38-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-62"><a href="#cb38-62" aria-hidden="true" tabindex="-1"></a>Extract the name (code) of your samples, and remove all the extra information from the filenames:</span>
<span id="cb38-63"><a href="#cb38-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-66"><a href="#cb38-66" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-67"><a href="#cb38-67" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"-"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">4</span>) </span>
<span id="cb38-68"><a href="#cb38-68" aria-hidden="true" tabindex="-1"></a><span class="co">#split the string when "-" is found and keep the 4th part of the string</span></span>
<span id="cb38-69"><a href="#cb38-69" aria-hidden="true" tabindex="-1"></a><span class="co">#we will get, for example, this: "A2S12R_S51_L001_R2_001.fastq.gz""</span></span>
<span id="cb38-70"><a href="#cb38-70" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(sample.names_raw), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb38-71"><a href="#cb38-71" aria-hidden="true" tabindex="-1"></a><span class="co">#removal of extra information (in this example, split sample.names_raw when "_" is found, and get the first part of the string)</span></span>
<span id="cb38-72"><a href="#cb38-72" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-73"><a href="#cb38-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-74"><a href="#cb38-74" aria-hidden="true" tabindex="-1"></a>We are going to rename the samples corresponding to the mock community, so that they satisfy the requirement of the functions to be applied in next steps:</span>
<span id="cb38-75"><a href="#cb38-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-78"><a href="#cb38-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-79"><a href="#cb38-79" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK1"</span>, <span class="st">"MOCK-1"</span>, sample.names)<span class="co">#replace "MOCK1" by "MOCK-1"</span></span>
<span id="cb38-80"><a href="#cb38-80" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK2"</span>, <span class="st">"MOCK-2"</span>, sample.names)</span>
<span id="cb38-81"><a href="#cb38-81" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span><span class="fu">gsub</span>(<span class="st">"MOCK3"</span>, <span class="st">"MOCK-3"</span>, sample.names)</span>
<span id="cb38-82"><a href="#cb38-82" aria-hidden="true" tabindex="-1"></a>sample.names <span class="co">#check that the name of our samples is OK</span></span>
<span id="cb38-83"><a href="#cb38-83" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-84"><a href="#cb38-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-85"><a href="#cb38-85" aria-hidden="true" tabindex="-1"></a><span class="fu"># 2. Check the quality of the sequencing</span></span>
<span id="cb38-86"><a href="#cb38-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-87"><a href="#cb38-87" aria-hidden="true" tabindex="-1"></a>There are different ways to check the quality of the reads:</span>
<span id="cb38-88"><a href="#cb38-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-89"><a href="#cb38-89" aria-hidden="true" tabindex="-1"></a><span class="fu">## a) Count the number of reads</span></span>
<span id="cb38-90"><a href="#cb38-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-91"><a href="#cb38-91" aria-hidden="true" tabindex="-1"></a>It would be nice to check whether we obtained enough reads from the sequencing service. Otherwise, we should ask the service to repeat the sequencing.</span>
<span id="cb38-92"><a href="#cb38-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-95"><a href="#cb38-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-96"><a href="#cb38-96" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb38-97"><a href="#cb38-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-98"><a href="#cb38-98" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)){</span>
<span id="cb38-99"><a href="#cb38-99" aria-hidden="true" tabindex="-1"></a>    raw_reads_count <span class="ot">=</span> <span class="fu">rbind</span>(raw_reads_count, </span>
<span id="cb38-100"><a href="#cb38-100" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb38-101"><a href="#cb38-101" aria-hidden="true" tabindex="-1"></a>  } <span class="co">#this loop counts the number of F reads by means of the ShortRead package</span></span>
<span id="cb38-102"><a href="#cb38-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-103"><a href="#cb38-103" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count)<span class="ot">=</span> sample.names <span class="co">#formatting of the output</span></span>
<span id="cb38-104"><a href="#cb38-104" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb38-105"><a href="#cb38-105" aria-hidden="true" tabindex="-1"></a>a<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count)</span>
<span id="cb38-106"><a href="#cb38-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-107"><a href="#cb38-107" aria-hidden="true" tabindex="-1"></a>raw_reads_count2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb38-108"><a href="#cb38-108" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnRs)){<span class="co">#do the same with R reads</span></span>
<span id="cb38-109"><a href="#cb38-109" aria-hidden="true" tabindex="-1"></a>  raw_reads_count2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count2, </span>
<span id="cb38-110"><a href="#cb38-110" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnRs[i])))</span>
<span id="cb38-111"><a href="#cb38-111" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb38-112"><a href="#cb38-112" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count2)<span class="ot">=</span> sample.names</span>
<span id="cb38-113"><a href="#cb38-113" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count2)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb38-114"><a href="#cb38-114" aria-hidden="true" tabindex="-1"></a>b<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count2),raw_reads_count2)</span>
<span id="cb38-115"><a href="#cb38-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-116"><a href="#cb38-116" aria-hidden="true" tabindex="-1"></a>a<span class="sc">==</span>b <span class="co">#check that we get the same number of F and R reads</span></span>
<span id="cb38-117"><a href="#cb38-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-118"><a href="#cb38-118" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)],</span>
<span id="cb38-119"><a href="#cb38-119" aria-hidden="true" tabindex="-1"></a>      <span class="fu">min</span>(raw_reads_count))<span class="co">#check which sample accounts for the lowest number of reads. Be careful if you keep the negative control of the sequencing</span></span>
<span id="cb38-120"><a href="#cb38-120" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)],</span>
<span id="cb38-121"><a href="#cb38-121" aria-hidden="true" tabindex="-1"></a>      <span class="fu">max</span>(raw_reads_count))<span class="co">#check which sample accounts for the highest number of reads.</span></span>
<span id="cb38-122"><a href="#cb38-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-123"><a href="#cb38-123" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count),</span>
<span id="cb38-124"><a href="#cb38-124" aria-hidden="true" tabindex="-1"></a>            <span class="at">file=</span><span class="st">"Num_raw_reads_16s.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb38-125"><a href="#cb38-125" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(raw_reads_count) <span class="co">#please, check the number of reads per sample</span></span>
<span id="cb38-126"><a href="#cb38-126" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-127"><a href="#cb38-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-128"><a href="#cb38-128" aria-hidden="true" tabindex="-1"></a>At this point, we have to check the number of raw reads per sample. The researcher has to verify whether this number is enough or not. It depends on the type of sample, among other parameters.</span>
<span id="cb38-129"><a href="#cb38-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-130"><a href="#cb38-130" aria-hidden="true" tabindex="-1"></a><span class="fu">## b) Inspect the length of the reads</span></span>
<span id="cb38-131"><a href="#cb38-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-132"><a href="#cb38-132" aria-hidden="true" tabindex="-1"></a>In our case, the genomics service followed 2x275 bp and 2x300 bp PE strategy, so most of the sequences are expected to have 275 or 300 bp.</span>
<span id="cb38-133"><a href="#cb38-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-136"><a href="#cb38-136" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-137"><a href="#cb38-137" aria-hidden="true" tabindex="-1"></a>reads<span class="ot">=</span>ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co">#save the reads in a new variable with the format of ShortRead package, which is a bit different from standard R variable</span></span>
<span id="cb38-138"><a href="#cb38-138" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">=</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co">#get the length of the reads</span></span>
<span id="cb38-139"><a href="#cb38-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-140"><a href="#cb38-140" aria-hidden="true" tabindex="-1"></a>counts<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb38-141"><a href="#cb38-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-142"><a href="#cb38-142" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb38-143"><a href="#cb38-143" aria-hidden="true" tabindex="-1"></a>  counts<span class="ot">=</span><span class="fu">rbind</span>(counts,</span>
<span id="cb38-144"><a href="#cb38-144" aria-hidden="true" tabindex="-1"></a>               <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width<span class="sc">==</span>uniques[i])))</span>
<span id="cb38-145"><a href="#cb38-145" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb38-146"><a href="#cb38-146" aria-hidden="true" tabindex="-1"></a>}<span class="co">#loop to count the number of reads of each length.  </span></span>
<span id="cb38-147"><a href="#cb38-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-148"><a href="#cb38-148" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">=</span>  <span class="fu">cbind</span>(uniques,counts)</span>
<span id="cb38-149"><a href="#cb38-149" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb38-150"><a href="#cb38-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-151"><a href="#cb38-151" aria-hidden="true" tabindex="-1"></a><span class="co">#check the histogram</span></span>
<span id="cb38-152"><a href="#cb38-152" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[,<span class="dv">1</span>],<span class="at">decreasing =</span> <span class="cn">TRUE</span>),]) <span class="co">#Most of the sequences should fall in expected sequence length</span></span>
<span id="cb38-153"><a href="#cb38-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-154"><a href="#cb38-154" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting</span></span>
<span id="cb38-155"><a href="#cb38-155" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main=</span><span class="st">"Forward length distribution"</span>,</span>
<span id="cb38-156"><a href="#cb38-156" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Sequence length"</span>, <span class="at">ylab=</span><span class="st">"Raw reads"</span>)</span>
<span id="cb38-157"><a href="#cb38-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-158"><a href="#cb38-158" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file=</span><span class="st">"Lenght_raw_reads_16S.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb38-159"><a href="#cb38-159" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-160"><a href="#cb38-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-161"><a href="#cb38-161" aria-hidden="true" tabindex="-1"></a><span class="fu">## c) Check the quality plots</span></span>
<span id="cb38-162"><a href="#cb38-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-163"><a href="#cb38-163" aria-hidden="true" tabindex="-1"></a>In this step, keep in mind the expected length of your amplicon. Then, check if the overlapping of F and R reads is possible considering the quality of the last nucleotides.</span>
<span id="cb38-164"><a href="#cb38-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-167"><a href="#cb38-167" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-168"><a href="#cb38-168" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])<span class="co">#select the specific samples you want to check, in this case 4 and 5</span></span>
<span id="cb38-169"><a href="#cb38-169" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb38-170"><a href="#cb38-170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-171"><a href="#cb38-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-172"><a href="#cb38-172" aria-hidden="true" tabindex="-1"></a><span class="fu"># 3. FIGARO tool</span></span>
<span id="cb38-173"><a href="#cb38-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-174"><a href="#cb38-174" aria-hidden="true" tabindex="-1"></a>We will use FIGARO to determine the best position of trimming in both F and R reads, and the best maximum expected errors for DADA2. **WARNING**: FIGARO does not run in R (not even in Windows!), so we will run it in python3</span>
<span id="cb38-175"><a href="#cb38-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-178"><a href="#cb38-178" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-179"><a href="#cb38-179" aria-hidden="true" tabindex="-1"></a>figFs <span class="ot">=</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnFs)) <span class="co">#create the directory where the output will be saved</span></span>
<span id="cb38-180"><a href="#cb38-180" aria-hidden="true" tabindex="-1"></a>figRs <span class="ot">=</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb38-181"><a href="#cb38-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-182"><a href="#cb38-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-183"><a href="#cb38-183" aria-hidden="true" tabindex="-1"></a>FIGARO does not work if all the samples are of different lengths. Thus, we have to cut the sequences so that this tool works. Do not panic because this cut is just made in this step. Then, we will work with the full-length reads</span>
<span id="cb38-184"><a href="#cb38-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-187"><a href="#cb38-187" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-188"><a href="#cb38-188" aria-hidden="true" tabindex="-1"></a>out.figaro<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, figFs, fnRs, figRs,<span class="at">compress=</span><span class="cn">TRUE</span>, </span>
<span id="cb38-189"><a href="#cb38-189" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">271</span>,<span class="dv">271</span>)) </span>
<span id="cb38-190"><a href="#cb38-190" aria-hidden="true" tabindex="-1"></a><span class="co">#select the position so that most of the sequences are considered </span></span>
<span id="cb38-191"><a href="#cb38-191" aria-hidden="true" tabindex="-1"></a><span class="co">#(check the histogram or the dataframe with the lengths of the reads to select this "pseudo"trimming positions)</span></span>
<span id="cb38-192"><a href="#cb38-192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-193"><a href="#cb38-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-196"><a href="#cb38-196" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-197"><a href="#cb38-197" aria-hidden="true" tabindex="-1"></a>figaro<span class="ot">=</span><span class="fu">system</span>((<span class="st">"python3 /home/programas/figaro/figaro/figaro.py  -i ~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/figaro -o ~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/figaro -a 426 -f 17 -r 21"</span>), <span class="at">intern=</span><span class="cn">TRUE</span>) </span>
<span id="cb38-198"><a href="#cb38-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-199"><a href="#cb38-199" aria-hidden="true" tabindex="-1"></a><span class="co">#indicate the path where the script "figaro.py" is located</span></span>
<span id="cb38-200"><a href="#cb38-200" aria-hidden="true" tabindex="-1"></a><span class="co">#-a, indicate the length of the amplicon, </span></span>
<span id="cb38-201"><a href="#cb38-201" aria-hidden="true" tabindex="-1"></a><span class="co">#f, indicate the length of primer F</span></span>
<span id="cb38-202"><a href="#cb38-202" aria-hidden="true" tabindex="-1"></a><span class="co">#r, indicate the length R  primer</span></span>
<span id="cb38-203"><a href="#cb38-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-204"><a href="#cb38-204" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(figaro) <span class="co">#select the best parameters proposed by FIGARO, here: Trimming position F,R: 248,236; maximum expected error F, R: 3,3.</span></span>
<span id="cb38-205"><a href="#cb38-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-206"><a href="#cb38-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-207"><a href="#cb38-207" aria-hidden="true" tabindex="-1"></a><span class="fu"># 4. Filter and trimming step</span></span>
<span id="cb38-208"><a href="#cb38-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-211"><a href="#cb38-211" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-212"><a href="#cb38-212" aria-hidden="true" tabindex="-1"></a>filtFs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))<span class="co">#create the "filtered" directory</span></span>
<span id="cb38-213"><a href="#cb38-213" aria-hidden="true" tabindex="-1"></a>filtRs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb38-214"><a href="#cb38-214" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-215"><a href="#cb38-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-218"><a href="#cb38-218" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-219"><a href="#cb38-219" aria-hidden="true" tabindex="-1"></a>out<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs, <span class="at">truncLen=</span><span class="fu">c</span>(<span class="dv">248</span>,<span class="dv">236</span>),</span>
<span id="cb38-220"><a href="#cb38-220" aria-hidden="true" tabindex="-1"></a>                     <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">3</span>), <span class="at">truncQ=</span><span class="dv">2</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>,</span>
<span id="cb38-221"><a href="#cb38-221" aria-hidden="true" tabindex="-1"></a>                     <span class="at">compress=</span><span class="cn">TRUE</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">minLen=</span><span class="dv">50</span>)</span>
<span id="cb38-222"><a href="#cb38-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-223"><a href="#cb38-223" aria-hidden="true" tabindex="-1"></a><span class="co">#truncLen: introduce the best trimming position proposed by FIGARO</span></span>
<span id="cb38-224"><a href="#cb38-224" aria-hidden="true" tabindex="-1"></a><span class="co">#maxN=0 : remove the reads with at least 1 ambiguity</span></span>
<span id="cb38-225"><a href="#cb38-225" aria-hidden="true" tabindex="-1"></a><span class="co">#trunQ=2: remove all the reads with one nucleotide with 2 or less quality value</span></span>
<span id="cb38-226"><a href="#cb38-226" aria-hidden="true" tabindex="-1"></a><span class="co">#maxEE: maximum expected error for F and R reads proposed by FIGARO</span></span>
<span id="cb38-227"><a href="#cb38-227" aria-hidden="true" tabindex="-1"></a><span class="co">#minLen=50: remove reads shorter than 50 bp</span></span>
<span id="cb38-228"><a href="#cb38-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-229"><a href="#cb38-229" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)</span>
<span id="cb38-230"><a href="#cb38-230" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-231"><a href="#cb38-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-232"><a href="#cb38-232" aria-hidden="true" tabindex="-1"></a>Our reads are partially filtered and trimmed (still have the primers!)</span>
<span id="cb38-233"><a href="#cb38-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-234"><a href="#cb38-234" aria-hidden="true" tabindex="-1"></a>Here you have the parameters we obtained for each sequencing runs:</span>
<span id="cb38-235"><a href="#cb38-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-236"><a href="#cb38-236" aria-hidden="true" tabindex="-1"></a>| Run      | Positions |       | maxEE |       |</span>
<span id="cb38-237"><a href="#cb38-237" aria-hidden="true" tabindex="-1"></a>|----------|-----------|-------|-------|-------|</span>
<span id="cb38-238"><a href="#cb38-238" aria-hidden="true" tabindex="-1"></a>|          | **F**     | **R** | **F** | **R** |</span>
<span id="cb38-239"><a href="#cb38-239" aria-hidden="true" tabindex="-1"></a>| **Run1** | 248       | 236   | 3     | 3     |</span>
<span id="cb38-240"><a href="#cb38-240" aria-hidden="true" tabindex="-1"></a>| **Run2** | 272       | 212   | 2     | 2     |</span>
<span id="cb38-241"><a href="#cb38-241" aria-hidden="true" tabindex="-1"></a>| **Run3** | 278       | 206   | 2     | 2     |</span>
<span id="cb38-242"><a href="#cb38-242" aria-hidden="true" tabindex="-1"></a>| **Run4** | 273       | 211   | 2     | 2     |</span>
<span id="cb38-243"><a href="#cb38-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-244"><a href="#cb38-244" aria-hidden="true" tabindex="-1"></a><span class="fu"># 5. Cutadapt: removal of the primers</span></span>
<span id="cb38-245"><a href="#cb38-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-246"><a href="#cb38-246" aria-hidden="true" tabindex="-1"></a>Our reads still have the primers (which are artificial sequences and can have ambiguities, *N* nucleotides). So, we have to remove them and discard all the sequences in which primers are not found (because if the sequencing run was ok, primers should had been found inside the reads). Let's use cutadapt tool, which does not run in R but in python:</span>
<span id="cb38-247"><a href="#cb38-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-250"><a href="#cb38-250" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-251"><a href="#cb38-251" aria-hidden="true" tabindex="-1"></a>FWD<span class="ot">=</span><span class="fu">c</span>(<span class="st">"CCTACGGGNBGCASCAG"</span>) <span class="co">#insert the sequences of the primers</span></span>
<span id="cb38-252"><a href="#cb38-252" aria-hidden="true" tabindex="-1"></a>REV<span class="ot">=</span><span class="fu">c</span>(<span class="st">"GACTACNVGGGTATCTAATCC"</span>) <span class="co">#here we have 2 degenerations (ambiguities)</span></span>
<span id="cb38-253"><a href="#cb38-253" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-254"><a href="#cb38-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-257"><a href="#cb38-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-258"><a href="#cb38-258" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">=</span> <span class="cf">function</span>(primer) {</span>
<span id="cb38-259"><a href="#cb38-259" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb38-260"><a href="#cb38-260" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">=</span><span class="fu">DNAString</span>(primer)  <span class="co">#package Biostrings does not work with characters but with strings, so let's change the class of the primers variables.</span></span>
<span id="cb38-261"><a href="#cb38-261" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">=</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), </span>
<span id="cb38-262"><a href="#cb38-262" aria-hidden="true" tabindex="-1"></a>              <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna), <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb38-263"><a href="#cb38-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co">#change from string to character</span></span>
<span id="cb38-264"><a href="#cb38-264" aria-hidden="true" tabindex="-1"></a>} <span class="co">#this function calculates all the possible orientations of the primers</span></span>
<span id="cb38-265"><a href="#cb38-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-266"><a href="#cb38-266" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">=</span> <span class="fu">allOrients</span>(FWD) <span class="co">#pass the function to F and R primers</span></span>
<span id="cb38-267"><a href="#cb38-267" aria-hidden="true" tabindex="-1"></a>REV.orients  <span class="ot">=</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb38-268"><a href="#cb38-268" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb38-269"><a href="#cb38-269" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb38-270"><a href="#cb38-270" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-271"><a href="#cb38-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-274"><a href="#cb38-274" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-275"><a href="#cb38-275" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">=</span><span class="cf">function</span>(primer, fn) {</span>
<span id="cb38-276"><a href="#cb38-276" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">=</span><span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb38-277"><a href="#cb38-277" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb38-278"><a href="#cb38-278" aria-hidden="true" tabindex="-1"></a>} <span class="co">#it counts the number of sequence in each orientation</span></span>
<span id="cb38-279"><a href="#cb38-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-280"><a href="#cb38-280" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),<span class="co">#check for instante, in sample number 4</span></span>
<span id="cb38-281"><a href="#cb38-281" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]),</span>
<span id="cb38-282"><a href="#cb38-282" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),</span>
<span id="cb38-283"><a href="#cb38-283" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]))</span>
<span id="cb38-284"><a href="#cb38-284" aria-hidden="true" tabindex="-1"></a><span class="co">#here we make a summary table. Please, visit DADA2 website for a better interpretation of the table.</span></span>
<span id="cb38-285"><a href="#cb38-285" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-286"><a href="#cb38-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-287"><a href="#cb38-287" aria-hidden="true" tabindex="-1"></a>Now yes, we are running cutadapt, although we have to adapt the reads previously for cutadapt. This tool needs some flags that have to be added to the reads. Visit the web of <span class="co">[</span><span class="ot">cutadapt</span><span class="co">](https://cutadapt.readthedocs.io/en/stable/guide.html "User guide of cutadapt")</span> for more information.</span>
<span id="cb38-288"><a href="#cb38-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-291"><a href="#cb38-291" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-292"><a href="#cb38-292" aria-hidden="true" tabindex="-1"></a>cutadapt  <span class="ot">=</span>  <span class="st">"/usr/bin/cutadapt"</span> <span class="co">#path to cutadapt </span></span>
<span id="cb38-293"><a href="#cb38-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-294"><a href="#cb38-294" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb38-295"><a href="#cb38-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-296"><a href="#cb38-296" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>) <span class="co">#create a directory where processed reads will be saved</span></span>
<span id="cb38-297"><a href="#cb38-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-298"><a href="#cb38-298" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb38-299"><a href="#cb38-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-300"><a href="#cb38-300" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb38-301"><a href="#cb38-301" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb38-302"><a href="#cb38-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-303"><a href="#cb38-303" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce arguments for cutadapt</span></span>
<span id="cb38-304"><a href="#cb38-304" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)  </span>
<span id="cb38-305"><a href="#cb38-305" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb38-306"><a href="#cb38-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-307"><a href="#cb38-307" aria-hidden="true" tabindex="-1"></a>R1.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>,FWD,<span class="st">"..."</span>, REV.RC) <span class="co">#adding the flags</span></span>
<span id="cb38-308"><a href="#cb38-308" aria-hidden="true" tabindex="-1"></a>R2.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-A"</span>,<span class="st">" "</span>,<span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb38-309"><a href="#cb38-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-310"><a href="#cb38-310" aria-hidden="true" tabindex="-1"></a><span class="co">#run cutadapt</span></span>
<span id="cb38-311"><a href="#cb38-311" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb38-312"><a href="#cb38-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>,<span class="st">"-m"</span>, <span class="dv">1</span>, <span class="st">"--discard-untrimmed"</span>, <span class="st">"-j"</span>,<span class="dv">0</span>, <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], filtFs[i], filtRs[i],<span class="st">"--report=minimal"</span>)) </span>
<span id="cb38-313"><a href="#cb38-313" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-314"><a href="#cb38-314" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-315"><a href="#cb38-315" aria-hidden="true" tabindex="-1"></a><span class="co">#-n 2: remove the primers</span></span>
<span id="cb38-316"><a href="#cb38-316" aria-hidden="true" tabindex="-1"></a><span class="co">#-m 1: remove empty sequences</span></span>
<span id="cb38-317"><a href="#cb38-317" aria-hidden="true" tabindex="-1"></a><span class="co">#-j 0: automatically detect number of cores</span></span>
<span id="cb38-318"><a href="#cb38-318" aria-hidden="true" tabindex="-1"></a><span class="co">#-0: output files</span></span>
<span id="cb38-319"><a href="#cb38-319" aria-hidden="true" tabindex="-1"></a><span class="co">#-i: input files. We use the filtered and trimmed reads</span></span>
<span id="cb38-320"><a href="#cb38-320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-321"><a href="#cb38-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-322"><a href="#cb38-322" aria-hidden="true" tabindex="-1"></a>Now, we will check whether cutadapt has removed all the primers. For that purpose, we pass the previously created function to check the number of times each primer appears in our dataset:</span>
<span id="cb38-323"><a href="#cb38-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-326"><a href="#cb38-326" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-327"><a href="#cb38-327" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),<span class="co">#Here we can indicate the number of the sample we want to check. In this case, sample number 4</span></span>
<span id="cb38-328"><a href="#cb38-328" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb38-329"><a href="#cb38-329" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb38-330"><a href="#cb38-330" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]))</span>
<span id="cb38-331"><a href="#cb38-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-332"><a href="#cb38-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-333"><a href="#cb38-333" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb38-334"><a href="#cb38-334" aria-hidden="true" tabindex="-1"></a>Note that in some cases, after the removal of the primers by cutadapt, the function *primerHits* still finds some primers in our dataset. Do no panic, these differences are normal because cutadapt and DADA2 functions are based on different algorithms.</span>
<span id="cb38-335"><a href="#cb38-335" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb38-336"><a href="#cb38-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-337"><a href="#cb38-337" aria-hidden="true" tabindex="-1"></a><span class="fu"># 6. DADA2: machine learning</span></span>
<span id="cb38-338"><a href="#cb38-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-339"><a href="#cb38-339" aria-hidden="true" tabindex="-1"></a>We are going to use DADA2 to denoise, infer the samples and to correct the expected errors by applying different functions of the package.</span>
<span id="cb38-340"><a href="#cb38-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-343"><a href="#cb38-343" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-344"><a href="#cb38-344" aria-hidden="true" tabindex="-1"></a><span class="co"># First, learn error rates</span></span>
<span id="cb38-345"><a href="#cb38-345" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">=</span><span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb38-346"><a href="#cb38-346" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">=</span><span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb38-347"><a href="#cb38-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-348"><a href="#cb38-348" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets plot the errors in each position</span></span>
<span id="cb38-349"><a href="#cb38-349" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-350"><a href="#cb38-350" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-351"><a href="#cb38-351" aria-hidden="true" tabindex="-1"></a><span class="co">#these plots show the possible errors made for each possible mutation (A&gt;C, A&gt;G, etc). Dots should be close to the red line.</span></span>
<span id="cb38-352"><a href="#cb38-352" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-353"><a href="#cb38-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-354"><a href="#cb38-354" aria-hidden="true" tabindex="-1"></a><span class="fu"># 7. DADA2: Sample inference</span></span>
<span id="cb38-355"><a href="#cb38-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-358"><a href="#cb38-358" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-359"><a href="#cb38-359" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">=</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err=</span>errF, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-360"><a href="#cb38-360" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">=</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err=</span>errR, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-361"><a href="#cb38-361" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">4</span>]]<span class="co">#check the inferred fourth sample (it's just an example). We will obtain the number of true sequence variants from the X number of unique sequences.</span></span>
<span id="cb38-362"><a href="#cb38-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-363"><a href="#cb38-363" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb38-364"><a href="#cb38-364" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">=</span> sample.names</span>
<span id="cb38-365"><a href="#cb38-365" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">=</span> sample.names</span>
<span id="cb38-366"><a href="#cb38-366" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-367"><a href="#cb38-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-368"><a href="#cb38-368" aria-hidden="true" tabindex="-1"></a><span class="fu"># 8. Overlap F and R sequences</span></span>
<span id="cb38-369"><a href="#cb38-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-370"><a href="#cb38-370" aria-hidden="true" tabindex="-1"></a>In this step we are going to overlap F (forward) and R (reverse) reads coming from the same sample.</span>
<span id="cb38-371"><a href="#cb38-371" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-374"><a href="#cb38-374" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-375"><a href="#cb38-375" aria-hidden="true" tabindex="-1"></a>mergers<span class="ot">=</span><span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-376"><a href="#cb38-376" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">4</span>]]) <span class="co">#check the output for sample number 4</span></span>
<span id="cb38-377"><a href="#cb38-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-378"><a href="#cb38-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-379"><a href="#cb38-379" aria-hidden="true" tabindex="-1"></a>Now, we are going to construct an amplicon sequence variant table (ASV table).</span>
<span id="cb38-380"><a href="#cb38-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-381"><a href="#cb38-381" aria-hidden="true" tabindex="-1"></a>Be careful, because we will get one ASV table per sequencing run. Thus, as we got 4 runs for bacterial amplicons, we will get 4 different ASV tables.</span>
<span id="cb38-382"><a href="#cb38-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-385"><a href="#cb38-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-386"><a href="#cb38-386" aria-hidden="true" tabindex="-1"></a>seqtab_run1<span class="ot">=</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb38-387"><a href="#cb38-387" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_run1) <span class="co">#indicates the number of samples (including mock community and negative controls) and the number of ASVs</span></span>
<span id="cb38-388"><a href="#cb38-388" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_run1, <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/BACTERIA/reads/seqtab_run1.rds"</span>) <span class="co">#save the seqtab in .rds format</span></span>
<span id="cb38-389"><a href="#cb38-389" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-390"><a href="#cb38-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-391"><a href="#cb38-391" aria-hidden="true" tabindex="-1"></a><span class="fu"># 9. Merge sequence tables from different runs</span></span>
<span id="cb38-392"><a href="#cb38-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-393"><a href="#cb38-393" aria-hidden="true" tabindex="-1"></a>As indicated before, in this step we are going to merge the sequence tables (seqtabs) coming from different runs into just one table. So, firstly, we have to load all of them to the current project. Please, note that all the seqtabs have to be in the same working directory!</span>
<span id="cb38-394"><a href="#cb38-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-397"><a href="#cb38-397" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-398"><a href="#cb38-398" aria-hidden="true" tabindex="-1"></a>seqtab_run1<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run1.rds"</span>)<span class="co">#load all the .rds files in the same directory</span></span>
<span id="cb38-399"><a href="#cb38-399" aria-hidden="true" tabindex="-1"></a>seqtab_run2<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run2.rds"</span>)</span>
<span id="cb38-400"><a href="#cb38-400" aria-hidden="true" tabindex="-1"></a>seqtab_run3<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run3.rds"</span>)</span>
<span id="cb38-401"><a href="#cb38-401" aria-hidden="true" tabindex="-1"></a>seqtab_run4<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run4.rds"</span>)</span>
<span id="cb38-402"><a href="#cb38-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-403"><a href="#cb38-403" aria-hidden="true" tabindex="-1"></a>mergedSeqTab<span class="ot">=</span><span class="fu">mergeSequenceTables</span>(seqtab_run1,seqtab_run2,seqtab_run3,seqtab_run4,</span>
<span id="cb38-404"><a href="#cb38-404" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">repeats=</span><span class="st">"sum"</span>)</span>
<span id="cb38-405"><a href="#cb38-405" aria-hidden="true" tabindex="-1"></a><span class="co">#with "sum" we indicate that we want to join and sum all the reads that are in samples with the same name. Why do we do that? Because in each sequencing run we have 3 replicates of the mock community, and we are going to sum the same replicates from different runs.</span></span>
<span id="cb38-406"><a href="#cb38-406" aria-hidden="true" tabindex="-1"></a><span class="co">#e.g., MOCK-1 (run1) + MOCK-1 (run2)+ MOCK-1 (run3)+ MOCK-1 (run4)</span></span>
<span id="cb38-407"><a href="#cb38-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-408"><a href="#cb38-408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-409"><a href="#cb38-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-410"><a href="#cb38-410" aria-hidden="true" tabindex="-1"></a><span class="fu"># 10. Chimeral removal</span></span>
<span id="cb38-411"><a href="#cb38-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-412"><a href="#cb38-412" aria-hidden="true" tabindex="-1"></a>As the sequencing process is based on PCRs, chimeric sequences are expect to appear. So, let's remove them.</span>
<span id="cb38-413"><a href="#cb38-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-416"><a href="#cb38-416" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-417"><a href="#cb38-417" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method=</span><span class="st">"consensus"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-418"><a href="#cb38-418" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co">#indicates the number of samples and ASVs (but not the number of sequences)</span></span>
<span id="cb38-419"><a href="#cb38-419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-420"><a href="#cb38-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-421"><a href="#cb38-421" aria-hidden="true" tabindex="-1"></a><span class="fu"># 11. Other steps: flter the ASVs by their length</span></span>
<span id="cb38-422"><a href="#cb38-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-423"><a href="#cb38-423" aria-hidden="true" tabindex="-1"></a>Let's check the number of ASVs and their length</span>
<span id="cb38-424"><a href="#cb38-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-427"><a href="#cb38-427" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-428"><a href="#cb38-428" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))  <span class="co">#Number of ASV of each length</span></span>
<span id="cb38-429"><a href="#cb38-429" aria-hidden="true" tabindex="-1"></a>  <span class="co">#first line: length of the sequences</span></span>
<span id="cb38-430"><a href="#cb38-430" aria-hidden="true" tabindex="-1"></a>  <span class="co">#second line: number of sequences of each length</span></span>
<span id="cb38-431"><a href="#cb38-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-432"><a href="#cb38-432" aria-hidden="true" tabindex="-1"></a><span class="co">#Our amplicon is of 426 bp in length, so most of the sequences would have that length</span></span>
<span id="cb38-433"><a href="#cb38-433" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-434"><a href="#cb38-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-435"><a href="#cb38-435" aria-hidden="true" tabindex="-1"></a>Calculate the number of sequences of each ASV that of each corresponding length:</span>
<span id="cb38-436"><a href="#cb38-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-439"><a href="#cb38-439" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-440"><a href="#cb38-440" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">=</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb38-441"><a href="#cb38-441" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb38-442"><a href="#cb38-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-443"><a href="#cb38-443" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">length=</span><span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count=</span>reads.per.seqlen)</span>
<span id="cb38-444"><a href="#cb38-444" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>count)) <span class="sc">+</span> <span class="fu">geom_col</span>()</span>
<span id="cb38-445"><a href="#cb38-445" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-446"><a href="#cb38-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-447"><a href="#cb38-447" aria-hidden="true" tabindex="-1"></a>Now, we have to filter the length of the ASVs. For that purpose, we have to keep in mind the data from the dataframe and the histogram. Choose those lengths in which we can retrieve enough sequences (<span class="sc">\&gt;</span>10000 sequences, if possible)</span>
<span id="cb38-448"><a href="#cb38-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-451"><a href="#cb38-451" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-452"><a href="#cb38-452" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> seqtab.nochim[,<span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab.nochim)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">401</span>,<span class="dv">428</span>)] <span class="co">#the consensus sequences of our ASVs will be from 401 to 428 nucleotides</span></span>
<span id="cb38-453"><a href="#cb38-453" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-454"><a href="#cb38-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-455"><a href="#cb38-455" aria-hidden="true" tabindex="-1"></a><span class="fu"># 12. Taxonomic classification</span></span>
<span id="cb38-456"><a href="#cb38-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-457"><a href="#cb38-457" aria-hidden="true" tabindex="-1"></a>We are going to use the database of the Ribosomal Database Project (RDP), specifically the trainset 19.</span>
<span id="cb38-458"><a href="#cb38-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-461"><a href="#cb38-461" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-462"><a href="#cb38-462" aria-hidden="true" tabindex="-1"></a>taxa_rdp2 <span class="ot">=</span><span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/home/databases/rdp_train_set_19_H.fa"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb38-463"><a href="#cb38-463" aria-hidden="true" tabindex="-1"></a><span class="co">#we selected the last version of the RDP-II training set v.19</span></span>
<span id="cb38-464"><a href="#cb38-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-465"><a href="#cb38-465" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">=</span> seqtab.nochim <span class="co">#edit the tables</span></span>
<span id="cb38-466"><a href="#cb38-466" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">=</span> <span class="fu">t</span>(ASV)</span>
<span id="cb38-467"><a href="#cb38-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-468"><a href="#cb38-468" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na <span class="ot">=</span> <span class="fu">apply</span>(taxa_rdp2,<span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[,<span class="sc">-</span><span class="dv">7</span>] <span class="co">#replace the "NA" values by "unclassified" in those case in which an ASV have not been identified at a specific taxonomic rank</span></span>
<span id="cb38-469"><a href="#cb38-469" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na<span class="ot">=</span>taxa_rdp_na[,<span class="sc">-</span>(<span class="dv">7</span><span class="sc">:</span><span class="dv">15</span>)]</span>
<span id="cb38-470"><a href="#cb38-470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-471"><a href="#cb38-471" aria-hidden="true" tabindex="-1"></a><span class="co">#we are going to modify a bit the tables so that the name of ASVs looks like "ASV00001" (in case we have 10000 ASVs or more). In case we have just 100 ASVs, they will have that code: "ASV001"</span></span>
<span id="cb38-472"><a href="#cb38-472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-473"><a href="#cb38-473" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">=</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb38-474"><a href="#cb38-474" aria-hidden="true" tabindex="-1"></a>names <span class="ot">=</span><span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co">#As many 0 as digits</span></span>
<span id="cb38-475"><a href="#cb38-475" aria-hidden="true" tabindex="-1"></a>ASV_names<span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb38-476"><a href="#cb38-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-477"><a href="#cb38-477" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_rdp_na,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASVt,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb38-478"><a href="#cb38-478" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-479"><a href="#cb38-479" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">=</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb38-480"><a href="#cb38-480" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb38-481"><a href="#cb38-481" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span>
<span id="cb38-482"><a href="#cb38-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-483"><a href="#cb38-483" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_table_classified_raw, <span class="at">file=</span><span class="st">"ASV_table_classified_raw_wirhMockwithPlastids.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb38-484"><a href="#cb38-484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-485"><a href="#cb38-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-486"><a href="#cb38-486" aria-hidden="true" tabindex="-1"></a>Now, we have our ASV table with the corresponding taxonomic classification. But, it still includes mitochondria and plastids from the host plants, the mock community and other invalid sequences.</span>
<span id="cb38-487"><a href="#cb38-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-488"><a href="#cb38-488" aria-hidden="true" tabindex="-1"></a><span class="fu"># 13. MOCK Community: setting the sequencing detection limit</span></span>
<span id="cb38-489"><a href="#cb38-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-490"><a href="#cb38-490" aria-hidden="true" tabindex="-1"></a>We included a mock community in order to establish the detection limit of the sequencing process. Now, we have to address this cut-off. For that purpose, we will run the function *MockCommunity* which will ask us whether specific microorganisms are included in the mock community. We have to answer accordingly, and the function will go on asking us until an ASV not included in the mock community is found. Its relative abundance in the whole dataset will be considered as the sequencing detection limit, since it should have not been detected if it was not included in the mock community (a false detection, so all the ASV whose relative abundance is below this cut-off, are not real ASVs). Then, we have to remove the mock community.</span>
<span id="cb38-491"><a href="#cb38-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-494"><a href="#cb38-494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-495"><a href="#cb38-495" aria-hidden="true" tabindex="-1"></a>ASV_filtered_MOCK<span class="ot">=</span><span class="fu">MockCommunity</span>(ASV_table_classified_raw,mock_composition,<span class="at">ASV_column =</span> <span class="st">"ASV_names"</span>)</span>
<span id="cb38-496"><a href="#cb38-496" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-497"><a href="#cb38-497" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-498"><a href="#cb38-498" aria-hidden="true" tabindex="-1"></a>::: callout-caution</span>
<span id="cb38-499"><a href="#cb38-499" aria-hidden="true" tabindex="-1"></a>Be careful with ASVs belonging to genus *Limosilactobacillus* (especially if their relative abundance is high). Possibly, they are missclassified and belong to genus *Bacillus*, which is a member of the MockCommunity here employed as well.</span>
<span id="cb38-500"><a href="#cb38-500" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb38-501"><a href="#cb38-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-502"><a href="#cb38-502" aria-hidden="true" tabindex="-1"></a><span class="fu"># 14. Removal of erronous sequences and taxonomy refinement</span></span>
<span id="cb38-503"><a href="#cb38-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-504"><a href="#cb38-504" aria-hidden="true" tabindex="-1"></a>This step is specially important if we are sequencing plant tissues (root, phyllosphere, etc), because with the primers employed here it is possible to amplify plant hosts' DNA, for example, that corresponding to mitochondria, chloroplasts, among others. Thus, let's remove all these undesired sequences. We will also remove those ASVs not classified even at kingdom level:</span>
<span id="cb38-505"><a href="#cb38-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-508"><a href="#cb38-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-509"><a href="#cb38-509" aria-hidden="true" tabindex="-1"></a>ASV_final<span class="ot">=</span>ASV_filtered_MOCK[(<span class="fu">which</span>(ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Streptophyta"</span>  <span class="sc">&amp;</span>ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Mitochondria"</span> <span class="sc">&amp;</span>  ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Chlorophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Genus<span class="sc">!=</span><span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Streptophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Bacillariophyta"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Mitochondria"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Class<span class="sc">!=</span><span class="st">"Chloroplast"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Order<span class="sc">!=</span><span class="st">"Chloroplast"</span> <span class="sc">&amp;</span>ASV_filtered_MOCK<span class="sc">$</span>Family<span class="sc">!=</span><span class="st">"Chloroplast"</span>  <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> ASV_filtered_MOCK<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"unclassified"</span>)),]</span>
<span id="cb38-510"><a href="#cb38-510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-511"><a href="#cb38-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-512"><a href="#cb38-512" aria-hidden="true" tabindex="-1"></a>Cyanobacterial sequences could be tricky, since they could be really close to plant sequences (chloroplasts). Thus, we should retrieve those sequences ascribed to the phylym *Cyanobacteriota*, and we then remove those that are not classified at class level. In our case, all the Cyanobacteria were chloroplasts (Phylum *Cyanobacteriota*, class Chloroplast), hence, they were removed in the previous step. But we encourage to make a BLAST at this point.</span>
<span id="cb38-513"><a href="#cb38-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-514"><a href="#cb38-514" aria-hidden="true" tabindex="-1"></a>We are going to check also the ASVs not classified at Phylum level</span>
<span id="cb38-515"><a href="#cb38-515" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-518"><a href="#cb38-518" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-519"><a href="#cb38-519" aria-hidden="true" tabindex="-1"></a>unclass_phy <span class="ot">=</span> ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"unclassified"</span>),] <span class="co">#save them into a new variable</span></span>
<span id="cb38-520"><a href="#cb38-520" aria-hidden="true" tabindex="-1"></a>seq_unclass <span class="ot">=</span> <span class="fu">as.list</span>(unclass_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb38-521"><a href="#cb38-521" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclass, <span class="at">names=</span>unclass_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"unclassified_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>) <span class="co">#write the sequences in fasta format</span></span>
<span id="cb38-522"><a href="#cb38-522" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-523"><a href="#cb38-523" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-524"><a href="#cb38-524" aria-hidden="true" tabindex="-1"></a>Now, we align by BLASTn these unclassified sequences against those held in the NCBI nt database, which is already downloaded in our PC.</span>
<span id="cb38-525"><a href="#cb38-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-528"><a href="#cb38-528" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-529"><a href="#cb38-529" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query unclassified_phylum.fas -db /home/databases/nt -out unclassified_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 5 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-530"><a href="#cb38-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-531"><a href="#cb38-531" aria-hidden="true" tabindex="-1"></a><span class="co">#we checked manually the output of BLAST and removed one ASV (ASV01487, unclassified at Phylum level with the RDP-II database) which was classified as *Musa* sp.</span></span>
<span id="cb38-532"><a href="#cb38-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-533"><a href="#cb38-533" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters<span class="ot">=</span>ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV01487"</span>)),]</span>
<span id="cb38-534"><a href="#cb38-534" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-535"><a href="#cb38-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-536"><a href="#cb38-536" aria-hidden="true" tabindex="-1"></a>Here we removed the sample corresponding to the negative control of the sequencing run (named "Cneg").</span>
<span id="cb38-537"><a href="#cb38-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-540"><a href="#cb38-540" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-541"><a href="#cb38-541" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters<span class="ot">=</span><span class="fu">subset</span>(ASV_final_all_filters, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(Cneg))</span>
<span id="cb38-542"><a href="#cb38-542" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-543"><a href="#cb38-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-544"><a href="#cb38-544" aria-hidden="true" tabindex="-1"></a>And finally, let's save the definitive bacterial ASV table:</span>
<span id="cb38-545"><a href="#cb38-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-548"><a href="#cb38-548" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-549"><a href="#cb38-549" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">" "</span><span class="ot">=</span><span class="fu">rownames</span>(ASV_final_all_filters),ASV_final_all_filters),<span class="at">file=</span><span class="st">"ASV_Bacterias_final.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb38-550"><a href="#cb38-550" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-551"><a href="#cb38-551" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-552"><a href="#cb38-552" aria-hidden="true" tabindex="-1"></a>We will also save the fasta file and perform a phylogenetic tree with **all the ASVs** because we will need a phylogenetic tree to perform further ecological analyses.</span>
<span id="cb38-553"><a href="#cb38-553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-556"><a href="#cb38-556" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-557"><a href="#cb38-557" aria-hidden="true" tabindex="-1"></a>seq_fasta <span class="ot">=</span> <span class="fu">as.list</span>(ASV_final_all_filters<span class="sc">$</span>ASV_seqs) </span>
<span id="cb38-558"><a href="#cb38-558" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_fasta, <span class="at">names=</span>ASV_final_all_filters<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"fasta_para_arbol16S.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb38-559"><a href="#cb38-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-560"><a href="#cb38-560" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-561"><a href="#cb38-561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-562"><a href="#cb38-562" aria-hidden="true" tabindex="-1"></a><span class="fu"># 15. Extra code</span></span>
<span id="cb38-563"><a href="#cb38-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-564"><a href="#cb38-564" aria-hidden="true" tabindex="-1"></a>As stated before, we will need a phylogenetic tree to calculate Weighted UniFrac distances among samples. Thus, here you have the code needed to calculate the tree (supposing you will calculate it in the same working path in which you have saved the fasta file *seq_fasta*)</span>
<span id="cb38-565"><a href="#cb38-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-566"><a href="#cb38-566" aria-hidden="true" tabindex="-1"></a>Firstly, we have to align all the sequences. For that purpose, we will use <span class="co">[</span><span class="ot">MAFFT software</span><span class="co">](https://mafft.cbrc.jp/alignment/server/index.html "Click to MAFFT website")</span></span>
<span id="cb38-567"><a href="#cb38-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-570"><a href="#cb38-570" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-571"><a href="#cb38-571" aria-hidden="true" tabindex="-1"></a>mafft <span class="ot">=</span> <span class="st">"/usr/bin/mafft"</span> <span class="co">#set MAFFT's path</span></span>
<span id="cb38-572"><a href="#cb38-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-573"><a href="#cb38-573" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(mafft, <span class="at">args=</span><span class="fu">c</span>(<span class="st">"--auto"</span>, <span class="st">"fasta_para_arbol16S.fas&gt;"</span>, <span class="st">"alignment"</span>))<span class="co">#after "--auto", set the name of the fasta file, followed by the name of the alignment.</span></span>
<span id="cb38-574"><a href="#cb38-574" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-575"><a href="#cb38-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-576"><a href="#cb38-576" aria-hidden="true" tabindex="-1"></a>Now, we are going to calculate the phylogenetic tree with <span class="co">[</span><span class="ot">FastTree MP tool</span><span class="co">](http://www.microbesonline.org/fasttree/ "Click to FastTree website")</span></span>
<span id="cb38-577"><a href="#cb38-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-580"><a href="#cb38-580" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb38-581"><a href="#cb38-581" aria-hidden="true" tabindex="-1"></a>FastTreeMP<span class="ot">=</span> <span class="st">"/home/programas/FastTreeMP/FastTreeMP"</span><span class="co">#set the path of the software</span></span>
<span id="cb38-582"><a href="#cb38-582" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-583"><a href="#cb38-583" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(FastTreeMP, <span class="at">args=</span><span class="st">"--version"</span> )</span>
<span id="cb38-584"><a href="#cb38-584" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(FastTreeMP, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"-gamma"</span>, <span class="st">"-nt"</span>, <span class="st">"-gtr"</span>, <span class="st">"-spr"</span>,<span class="dv">4</span> ,<span class="st">"-mlacc"</span>, <span class="dv">2</span>, <span class="st">"-slownni"</span>, <span class="st">"&lt;alignment&gt;"</span>, <span class="st">"tree"</span>))</span>
<span id="cb38-585"><a href="#cb38-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-586"><a href="#cb38-586" aria-hidden="true" tabindex="-1"></a><span class="co">#Our tree is Gamma20-based likelihood tree, based on the input alignment named "alignment", and the output tree is named "tree".</span></span>
<span id="cb38-587"><a href="#cb38-587" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb38-588"><a href="#cb38-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-589"><a href="#cb38-589" aria-hidden="true" tabindex="-1"></a>Click <span class="co">[</span><span class="ot">here</span><span class="co">](Bacteria_data_analysis.qmd "Bacterial data analyses")</span> to visit the scripts needed for the ecological analyses.</span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>