<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Banana - Fungi</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Banana</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-processing-of-miseq-reads" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Processing of MiSeq reads</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-processing-of-miseq-reads">    
        <li>
    <a class="dropdown-item" href="./Bacteria_reads_processing.html">
 <span class="dropdown-text">16S rRNA reads</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_reads_processing.html">
 <span class="dropdown-text">ITS2 reads</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-ecological-analyses" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Ecological analyses</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-ecological-analyses">    
        <li>
    <a class="dropdown-item" href="./Bacteria_data_analysis.html">
 <span class="dropdown-text">Bacterial dataset</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_data_analysis.html">
 <span class="dropdown-text">Fungal dataset</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/anitalasa"> <i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
<span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://orcid.org/0000-0003-3783-7157"> 
<span class="menu-text"><img src="images/orcid.png" title="ORCID" alt="ORCID" style="height:20px;"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://grupos.eez.csic.es/mae/"> 
<span class="menu-text"><img src="images/internet.png" title="web research group" alt="web research group" style="height:20px;"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#formatting-the-name-of-the-samples" id="toc-formatting-the-name-of-the-samples" class="nav-link active" data-scroll-target="#formatting-the-name-of-the-samples">1. Formatting the name of the samples</a></li>
  <li><a href="#check-the-quality-of-the-sequencing" id="toc-check-the-quality-of-the-sequencing" class="nav-link" data-scroll-target="#check-the-quality-of-the-sequencing">2. Check the quality of the sequencing</a>
  <ul class="collapse">
  <li><a href="#a-count-the-number-of-reads" id="toc-a-count-the-number-of-reads" class="nav-link" data-scroll-target="#a-count-the-number-of-reads">a) Count the number of reads</a></li>
  <li><a href="#b-inspect-the-length-of-the-reads" id="toc-b-inspect-the-length-of-the-reads" class="nav-link" data-scroll-target="#b-inspect-the-length-of-the-reads">b) Inspect the length of the reads</a></li>
  <li><a href="#c-check-the-quality-plots" id="toc-c-check-the-quality-plots" class="nav-link" data-scroll-target="#c-check-the-quality-plots">c) Check the quality plots</a></li>
  </ul></li>
  <li><a href="#filter-and-trimming-step" id="toc-filter-and-trimming-step" class="nav-link" data-scroll-target="#filter-and-trimming-step">3. Filter and trimming step</a></li>
  <li><a href="#cutadapt-removal-of-the-primers" id="toc-cutadapt-removal-of-the-primers" class="nav-link" data-scroll-target="#cutadapt-removal-of-the-primers">4. Cutadapt: removal of the primers</a></li>
  <li><a href="#dada2-machine-learning" id="toc-dada2-machine-learning" class="nav-link" data-scroll-target="#dada2-machine-learning">5. DADA2: machine learning</a></li>
  <li><a href="#dada2-sample-inference" id="toc-dada2-sample-inference" class="nav-link" data-scroll-target="#dada2-sample-inference">6. DADA2: Sample inference</a></li>
  <li><a href="#merge-sequences" id="toc-merge-sequences" class="nav-link" data-scroll-target="#merge-sequences">7. Merge sequences</a></li>
  <li><a href="#merge-sequence-tables-from-different-runs" id="toc-merge-sequence-tables-from-different-runs" class="nav-link" data-scroll-target="#merge-sequence-tables-from-different-runs">8. Merge sequence tables from different runs</a></li>
  <li><a href="#chimeral-removal" id="toc-chimeral-removal" class="nav-link" data-scroll-target="#chimeral-removal">9. Chimeral removal</a></li>
  <li><a href="#other-steps-flter-the-asvs-by-their-length" id="toc-other-steps-flter-the-asvs-by-their-length" class="nav-link" data-scroll-target="#other-steps-flter-the-asvs-by-their-length">10. Other steps: flter the ASVs by their length</a></li>
  <li><a href="#taxonomic-classification" id="toc-taxonomic-classification" class="nav-link" data-scroll-target="#taxonomic-classification">11. Taxonomic classification</a></li>
  <li><a href="#mock-community-setting-the-sequencing-detection-limit" id="toc-mock-community-setting-the-sequencing-detection-limit" class="nav-link" data-scroll-target="#mock-community-setting-the-sequencing-detection-limit">12. MOCK Community: setting the sequencing detection limit</a></li>
  <li><a href="#removal-of-erronous-sequences-and-taxonomy-refinement" id="toc-removal-of-erronous-sequences-and-taxonomy-refinement" class="nav-link" data-scroll-target="#removal-of-erronous-sequences-and-taxonomy-refinement">13. Removal of erronous sequences and taxonomy refinement</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Fungi</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>First of all, we will install and load of the required packages and libraries</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"nuriamw/micro4all"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(micro4all)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>As you can see, we are going to use different functions of <a href="https://benjjneb.github.io/dada2/tutorial.html" title="Click to DADA2 website">DADA2 package</a>, so it will be nice if you read all the documentation regarding this package.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>The steps 1 to 7 must be performed for each sequencing run. If you are working with different sequencing runs, you have to join all the sequences tables into one object (step 8). Once the joined element is obtained, steps 9-last step have to be applied to the joined object.</p>
</div>
</div>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you have included a Mock community which does not include fungal sequences (or a very small number of fungal sequences), I suggest you to process first the 16S rRNA dataset.</p>
</div>
</div>
<section id="formatting-the-name-of-the-samples" class="level1">
<h1>1. Formatting the name of the samples</h1>
<p>Now, we will start! Set the working directory and specify the path where you will be working:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>path<span class="ot">=</span> <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/310523_ITS/reads"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path) <span class="co">#Check that all files are here included (script + fastq files)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Sort F and R reads separately and save them into two variables:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co">#our filenames have the format "NGS015-23-ITS2-A2S12R_S51_L001_R2_001.fastq.gz"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Extract the name (code) of our samples, and remove all the extra information from the filenames:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>) </span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#split the string when "_" is found and keep the 4th part of the string</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we will get, for example, this: NGS015-23-ITS2-A2S12R</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span> <span class="fu">gsub</span>(<span class="at">patter =</span> <span class="st">"NGS015-23-ITS2-"</span>, <span class="at">replacement =</span> <span class="st">""</span>, sample.names)<span class="co">#here we replace the extra information by nothing (no characters)</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note that in this case, we have modified the sample names by employing a different way from that used for bacterial dataset. However, the result is exactly the same.</p>
</div>
</div>
</section>
<section id="check-the-quality-of-the-sequencing" class="level1">
<h1>2. Check the quality of the sequencing</h1>
<p>There are different ways to check the quality of the reads:</p>
<section id="a-count-the-number-of-reads" class="level2">
<h2 class="anchored" data-anchor-id="a-count-the-number-of-reads">a) Count the number of reads</h2>
<p>It would be nice to check whether we obtained enough reads from the sequencing service. Otherwise, we should ask for the service to repeat the sequencing.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)){</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>    raw_reads_count <span class="ot">=</span> <span class="fu">rbind</span>(raw_reads_count, </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  } <span class="co">#this loop counts the number of F reads by means of the ShortRead package</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count)<span class="ot">=</span> sample.names <span class="co">#formatting of the output</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>a<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count)</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>raw_reads_count2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnRs)){<span class="co">#do the same with R reads</span></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  raw_reads_count2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count2, </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnRs[i])))</span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count2)<span class="ot">=</span> sample.names</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count2)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>b<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count2),raw_reads_count2)</span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>a<span class="sc">==</span>b <span class="co">#check that we get the same number of F and R reads</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)],</span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">min</span>(raw_reads_count))<span class="co">#check which sample accounts for the lowest number of reads. Be careful if you keep the negative control of the sequencing</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)],</span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a>      <span class="fu">max</span>(raw_reads_count))<span class="co">#check which sample accounts for the highest number of reads.</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count),</span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a>            <span class="at">file=</span><span class="st">"Num_raw_reads_ITS.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(raw_reads_count) <span class="co">#please, check the number of reads per sample</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>At this point, we have to check the number of raw reads per sample. The researcher has to verify whether this number is enough or not. It depends on the type of sample, among other parameters.</p>
</section>
<section id="b-inspect-the-length-of-the-reads" class="level2">
<h2 class="anchored" data-anchor-id="b-inspect-the-length-of-the-reads">b) Inspect the length of the reads</h2>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Remember that ITS2 varies in length among different fungi. We have to keep this in mind along the analysis, since we have to skip several steps based on the length of the reads (comparing to the processing of 16S <em>rRNA</em> reads).</p>
</div>
</div>
<p>In spite of that indicated in the above callout, we will inspect the length of the reads just to confirm the genomics service has performed a good job.</p>
<p>In our case, the genomics service followed a 2x300 bp PE strategy.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>reads<span class="ot">=</span>ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co">#save the reads in a new variable with the format of ShortRead package, which is a bit different from standard R variable</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">=</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co">#get the length of the reads</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>counts<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  counts<span class="ot">=</span><span class="fu">rbind</span>(counts,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>               <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width<span class="sc">==</span>uniques[i])))</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>}<span class="co">#specific loop to count the number of reads of each length.  </span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">=</span>  <span class="fu">cbind</span>(uniques,counts)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co">#check the histogram</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[,<span class="dv">1</span>],<span class="at">decreasing =</span> <span class="cn">TRUE</span>),]) <span class="co">#Most of the sequences should fall in expected sequence length</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main=</span><span class="st">"Forward length distribution"</span>,</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Sequence length"</span>, <span class="at">ylab=</span><span class="st">"Raw reads"</span>)</span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file=</span><span class="st">"Lenght_raw_reads_ITS.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="c-check-the-quality-plots" class="level2">
<h2 class="anchored" data-anchor-id="c-check-the-quality-plots">c) Check the quality plots</h2>
<p>Check if the overlapping of F and R reads is possible considering the quality of the last nucleotides. Be careful with the quality of R reads (which tends to be worse than that of F reads, especially in the latest part of the reads)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])<span class="co">#select the specific samples you want to check, in this case 4 and 5</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="filter-and-trimming-step" class="level1">
<h1>3. Filter and trimming step</h1>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>If you come from the processing of bacterial reads, you would have notice that here we have skipped the step of FIGARO. That tool needs that all the reads are of the same length, so we cannot use it for fungal (ITS2) dataset.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>filtFs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))<span class="co">#create the "filtered" directory</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>filtRs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Since we cannot use FIGARO, we have to test different parameters in the following filtering and trimming step. It is recommended to achieve a compromise between the percentage of retained sequences and their quality. So, run the following command indicating specific values of maxEE according to the quality plots of F and R reads previously visualized in the step 2.c). Write down the results obtained for that maxEE, and then re-run with another maxEE values, until you reach to the best solution.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>out<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs, <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">truncQ=</span><span class="dv">2</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>, <span class="at">compress=</span><span class="cn">TRUE</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">minLen=</span><span class="dv">50</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#maxN=0 : remove the reads with at least 1 ambiguity</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#trunQ=2: remove all the reads with one nucleotide with 2 or less quality value</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#maxEE: maximum expected error for F and R reads</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#minLen=50: remove reads shorter than 50 bp</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)<span class="co">#important! Check the results</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In our case, we selected:</p>
<table class="table">
<thead>
<tr class="header">
<th>Sequencing run</th>
<th>Positions</th>
<th></th>
<th>maxEE</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td><strong>F</strong></td>
<td><strong>R</strong></td>
<td><strong>F</strong></td>
<td><strong>R</strong></td>
</tr>
<tr class="even">
<td><strong>Run5</strong></td>
<td></td>
<td></td>
<td>3</td>
<td>4</td>
</tr>
<tr class="odd">
<td><strong>Run6</strong></td>
<td></td>
<td></td>
<td>3</td>
<td>4</td>
</tr>
<tr class="even">
<td><strong>Run7</strong></td>
<td></td>
<td></td>
<td>4</td>
<td>6</td>
</tr>
</tbody>
</table>
<p>Our reads are partially filtered and trimmed (still have the primers!)</p>
</section>
<section id="cutadapt-removal-of-the-primers" class="level1">
<h1>4. Cutadapt: removal of the primers</h1>
<p>Our reads still have the primers (which are artificial sequences and can have ambiguities). So, we have to remove them and discard all the sequences in which primers are not found (because if the sequencing run was ok, primers should had been found inside the reads). Lets use cutadapt tool, which does not run in R but in python:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">=</span> <span class="st">"GTGARTCATCGAATCTTTG"</span> <span class="co">#sequence of F primer</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">=</span> <span class="st">"TCCTCCGCTTATTGATATGC"</span> <span class="co">#sequence of R primer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">=</span> <span class="cf">function</span>(primer) {</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">=</span><span class="fu">DNAString</span>(primer)  <span class="co">#package Biostrings does not work with characters but with strings, so let's change the class of the primers variables.</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">=</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna), <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co">#change from string to character</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>} <span class="co">#this function calculates all the possible orientations of the primers</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">=</span> <span class="fu">allOrients</span>(FWD) <span class="co">#pass the function to F and R primers</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>REV.orients  <span class="ot">=</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>REV.orients</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">=</span><span class="cf">function</span>(primer, fn) {</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">=</span><span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>} <span class="co">#it counts the number of sequence in each orientation</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),<span class="co">#check for instante, in sample number 4</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]),</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]))</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="co">#here we make a summary table. Please, visit DADA2 website for a better interpretation of the table.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now yes, we are running cutadapt, although we have to adapt the reads previously for cutadapt. This tool need some flags that have to be added to the reads. Visit the web of <a href="https://cutadapt.readthedocs.io/en/stable/guide.html" title="User guide of cutadapt">cutadapt</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>cutadapt  <span class="ot">=</span>  <span class="st">"/usr/bin/cutadapt"</span> <span class="co">#path to cutadapt </span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>) <span class="co">#create a directory where processed reads will be saved</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(filtFs))<span class="co">#if you came from the processing of 16S rRNA reads, be careful here. We have to use "filtFs" instead of "fnFs" because we have already performed the filtering and trimming steps.</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(filtRs))<span class="co">#the same comment here</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce arguments for cutadapt</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)  </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>R1.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>,FWD,<span class="st">"..."</span>, REV.RC) <span class="co">#adding the flags</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>R2.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-A"</span>,<span class="st">" "</span>,<span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a><span class="co">#run cutadapt</span></span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>,<span class="st">"-m"</span>, <span class="dv">1</span>, <span class="st">"--discard-untrimmed"</span>, <span class="st">"-j"</span>,<span class="dv">0</span>, <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], filtFs[i], filtRs[i],<span class="st">"--report=minimal"</span>)) </span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a><span class="co">#-n 2: remove the primers</span></span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a><span class="co">#-m 1: remove empty sequences</span></span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a><span class="co">#-j 0: automatically detect number of cores</span></span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a><span class="co">#-0: output files</span></span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="co">#-i: input files. We use the filtered and trimmed reads</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we check whether cutadapt has removed all the primers. For that purpose, we pass the previously created function to check the number of times each primer appears in our dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),<span class="co">#Here we can indicate the number of the sample we want to check. In this case, sample number 4</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Note that in some cases, after the removal of the primers by cutadapt, the function primerHits still finds some primers in our dataset. Do no pannic; these differences are normal because cutadapt and that DADA2 functions are based on different algorithms.</p>
</div>
</div>
</section>
<section id="dada2-machine-learning" class="level1">
<h1>5. DADA2: machine learning</h1>
<p>We are going to use DADA2 to denoise, infer the samples and to correct the expected errors by applying different functions of the package</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First, learn error rates</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">=</span><span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">=</span><span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets plot the errors in each position</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">#these plots show the possible errors made for each possible mutation (A&gt;C, A&gt;G, etc). Dots should be close to the red line.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="dada2-sample-inference" class="level1">
<h1>6. DADA2: Sample inference</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">=</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err=</span>errF, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">=</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err=</span>errR, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">4</span>]]<span class="co">#check the inferred fourth sample (it's just an example). We will obtain the number of true sequence variants from the X number of unique sequences.</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">=</span> sample.names</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">=</span> sample.names</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merge-sequences" class="level1">
<h1>7. Merge sequences</h1>
<p>In this step we are going to overlap F and R reads coming from the sample sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>mergers<span class="ot">=</span><span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">4</span>]]) <span class="co">#check the output for sample number 4</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we are going to construct an amplicon sequence variant table (ASV table).</p>
<p>Be careful, because we will get one ASV table per sequencing run. Thus, as we got 3 runs for fungal amplicons, we will get 3 different ASV table.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>seqtab_run5<span class="ot">=</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_run5) <span class="co">#indicates the number of samples (including negative controls) and the number of ASVs</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_run5, <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/310523_ITS/reads/seqtab_run5.rds"</span>) <span class="co">#save the seqtab in .rds format</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="merge-sequence-tables-from-different-runs" class="level1">
<h1>8. Merge sequence tables from different runs</h1>
<p>As indicated before, in this step we are going to merge the sequence tables coming from different runs. So, firstly, we have to load all of them to the current project. Please, note that all the seqtabs have to be in the same working directory!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>seqtab_run5<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run5.rds"</span>)<span class="co">#load all the .rds files in the same directory</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>seqtab_run6<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run6rds"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>seqtab_run7<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run7rds"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>mergedSeqTab<span class="ot">=</span><span class="fu">mergeSequenceTables</span>(seqtab_run5,seqtab_run6,seqtab_run7,<span class="at">repeats=</span><span class="st">"sum"</span>)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co">#with "sum" we indicate that we want to join and sum all the reads that are in samples with the same name. Why do we do that? </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="chimeral-removal" class="level1">
<h1>9. Chimeral removal</h1>
<p>As the sequencing process is based on PCRs, chimeric sequences are expect to appear. So, lets remove them</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method=</span><span class="st">"consensus"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)<span class="co">#if you just are working on one dataset (coming from just one sequencing run, instead of "mergedSeqTab", you have to write "seqt_tab" or the name of your seqtab)</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co">#indicates the number of samples and ASVs (but not the number of sequences)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="other-steps-flter-the-asvs-by-their-length" class="level1">
<h1>10. Other steps: flter the ASVs by their length</h1>
<p>Lets check the number of ASVs and their length</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))  <span class="co">#Number of ASV of each length</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="co">#first line: length of the sequences</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="co">#second line: number of sequences of each length</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Calculate the number of sequences of each ASV that that each length:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">=</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">length=</span><span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count=</span>reads.per.seqlen)</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>count)) <span class="sc">+</span> <span class="fu">geom_col</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxonomic-classification" class="level1">
<h1>11. Taxonomic classification</h1>
<p>We are going to classify our ITS2 amplicons against <a href="https://doi.org/10.15156/BIO/2938067" title="Click to the paper">UNITE</a> v.9.0 databse</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>taxa_unite <span class="ot">=</span><span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/home/databases/sh_general_release_dynamic_25.07.2023.fasta"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>)<span class="co">#indicate the path were your database is located</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">=</span> seqtab.nochim <span class="co">#edit the tables</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">=</span> <span class="fu">t</span>(ASV)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>taxa_na <span class="ot">=</span> <span class="fu">apply</span>(taxa_unite,<span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[,<span class="sc">-</span><span class="dv">7</span>] <span class="co">#replace the "NA" values by "unclassified" in those case in which an ASV have not been identified at a specific taxonomic rank</span></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na<span class="ot">=</span>taxa_rdp_na[,<span class="sc">-</span>(<span class="dv">7</span><span class="sc">:</span><span class="dv">15</span>)]</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co">#we are going to modify a bit the tables so that the name of ASVs looks like "ASV00001" (in case we have 10000 ASVs or more). In case we have just 100 ASVs, they will have that code: "ASV001"</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">=</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>names <span class="ot">=</span><span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co">#As many 0 as digits</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>ASV_names<span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_na,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASVt,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">=</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb23-18"><a href="#cb23-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb23-19"><a href="#cb23-19" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span>
<span id="cb23-20"><a href="#cb23-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-21"><a href="#cb23-21" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_table_classified_raw, <span class="at">file=</span><span class="st">"ASV_table_classified_raw_wirhMockwithPlastids.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, we have our ASV table with the corresponding taxonomic classification. But, it still potentially includes some sequences that could be erroneous.</p>
</section>
<section id="mock-community-setting-the-sequencing-detection-limit" class="level1">
<h1>12. MOCK Community: setting the sequencing detection limit</h1>
<p>The Mock Community we used in our sequencing runs just include sequences of two fungi. Thus, we will consider that the detection limit previously established for bacterial dataset is the same as for fungi. Ours was 0.001207% of the sequences. Lets use it:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>ASV_sums<span class="ot">=</span><span class="fu">rowSums</span>(ASV_table_classified_raw[,<span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(ASV_table_classified_raw)])</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>sum.total<span class="ot">=</span><span class="fu">sum</span>(ASV_sums)<span class="co"># Get the total number of sequences</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>nseq_cutoff<span class="ot">=</span>(<span class="fl">0.001207</span><span class="sc">/</span><span class="dv">100</span>)<span class="sc">*</span>sum.total<span class="co"># Apply the percentage to sequence number</span></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Now, filter the table accordingly:</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>ASV_filtered<span class="ot">=</span>ASV_table_classified_raw[<span class="fu">which</span>(ASV_sums<span class="sc">&gt;</span>nseq_cutoff),] <span class="co">#we are retaining just those ASVs accounting for more sequences that that determined by the cut-off</span></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>ASV_table<span class="ot">=</span>ASV_filtered[<span class="fu">order</span>(ASV_filtered[[<span class="st">"ASV_names"</span>]]),]<span class="co"># Sort table in ascending order of ASV names</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="removal-of-erronous-sequences-and-taxonomy-refinement" class="level1">
<h1>13. Removal of erronous sequences and taxonomy refinement</h1>
<p>Lets remove all the undesired sequences (if we are working with plant endophytes, we will have retained plant sequences). We will also remove those ASVs not classified even at kingdom level.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>ASV_final<span class="ot">=</span>ASV_table[(<span class="fu">which</span>(ASV_table<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"unclassified"</span>)),] <span class="co">#remove all the sequences not classified at Kingdom level</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Sometimes, when we have worked with plant tissues, we can find some fungi that are adscribed to <em>incertae sedis</em> group (missclassified) even at phylum level. They tend to be sequences of the plant host. So, lets verify it and remove then in case we find these sequences:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>incertae_phy<span class="ot">=</span>ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"p__Fungi_phy_Incertae_sedis"</span>),] <span class="co">#UNITE database writes "p__" prior to the name of the phyla</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>seq_incertae<span class="ot">=</span><span class="fu">as.list</span>(incertae_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_incertae, <span class="at">names=</span>incertae_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"incertaesedis_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)<span class="co">#save the fasta file of these ASVs</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Let's perform a comparison of these sequences against the NCBI GenBank database by BLASTn</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query incertaesedis_phylum.fas -db /home/databases/nt -out incertae_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co">#The output of the BLAST ("incertae_phylum_hits.txt") have to be manually checked.</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="co">#In our case, all the sequences are related to *Musa* spp. and other eukaryots that are not fungi, so we are going to remove all these sequences:</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>ASV_final2 <span class="ot">=</span> ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">!=</span><span class="st">"p__Fungi_phy_Incertae_sedis"</span>)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Repeat the same with the unclassified sequences at phylum level:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>unclassified_phy<span class="ot">=</span>ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"unclassified"</span>),]</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>seq_unclassified<span class="ot">=</span> <span class="fu">as.list</span>(unclassified_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclassified, <span class="at">names=</span>unclassified_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"unclassified_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query unclassified_phylum.fas -db /home/databases/nt -out unclassified_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 5 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a><span class="co">#We found a lot of missclassified ASVs at phylum, so we are going to remove them especifically:</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>ASV_final3 <span class="ot">=</span> ASV_final2[(<span class="fu">which</span>(ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0004"</span><span class="sc">&amp;</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0012"</span><span class="sc">&amp;</span></span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0109"</span><span class="sc">&amp;</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0140"</span><span class="sc">&amp;</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0269"</span><span class="sc">&amp;</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0627"</span><span class="sc">&amp;</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0654"</span><span class="sc">&amp;</span></span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1112"</span><span class="sc">&amp;</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0938"</span><span class="sc">&amp;</span></span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0911"</span><span class="sc">&amp;</span></span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0731"</span><span class="sc">&amp;</span></span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0844"</span><span class="sc">&amp;</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0923"</span><span class="sc">&amp;</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1080"</span><span class="sc">&amp;</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1096"</span><span class="sc">&amp;</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1147"</span><span class="sc">&amp;</span></span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1291"</span><span class="sc">&amp;</span></span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1321"</span><span class="sc">&amp;</span></span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1335"</span>)),]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Eventually, save the definitive ASV table:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">" "</span><span class="ot">=</span><span class="fu">rownames</span>(ASV_final3),ASV_final3),<span class="at">file=</span><span class="st">"ASV_Hongos_FINAL.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb29" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Fungi"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>First of all, we will install and load of the required packages and libraries</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"devtools"</span>)</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"nuriamw/micro4all"</span>)</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(micro4all)</span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>As you can see, we are going to use different functions of <span class="co">[</span><span class="ot">DADA2 package</span><span class="co">](https://benjjneb.github.io/dada2/tutorial.html "Click to DADA2 website")</span>, so it will be nice if you read all the documentation regarding this package.</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>::: callout-important</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a>The steps 1 to 7 must be performed for each sequencing run. If you are working with different sequencing runs, you have to join all the sequences tables into one object (step 8). Once the joined element is obtained, steps 9-last step have to be applied to the joined object.</span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a>::: callout-warning</span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a>If you have included a Mock community which does not include fungal sequences (or a very small number of fungal sequences), I suggest you to process first the 16S rRNA dataset.</span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a><span class="fu"># 1. Formatting the name of the samples</span></span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-47"><a href="#cb29-47" aria-hidden="true" tabindex="-1"></a>Now, we will start! Set the working directory and specify the path where you will be working:</span>
<span id="cb29-48"><a href="#cb29-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a>path<span class="ot">=</span> <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/310523_ITS/reads"</span></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path) <span class="co">#Check that all files are here included (script + fastq files)</span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a>Sort F and R reads separately and save them into two variables:</span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern=</span><span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="co">#our filenames have the format "NGS015-23-ITS2-A2S12R_S51_L001_R2_001.fastq.gz"</span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a>Extract the name (code) of our samples, and remove all the extra information from the filenames:</span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">=</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>) </span>
<span id="cb29-72"><a href="#cb29-72" aria-hidden="true" tabindex="-1"></a><span class="co">#split the string when "_" is found and keep the 4th part of the string</span></span>
<span id="cb29-73"><a href="#cb29-73" aria-hidden="true" tabindex="-1"></a><span class="co">#we will get, for example, this: NGS015-23-ITS2-A2S12R</span></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a>sample.names<span class="ot">=</span> <span class="fu">gsub</span>(<span class="at">patter =</span> <span class="st">"NGS015-23-ITS2-"</span>, <span class="at">replacement =</span> <span class="st">""</span>, sample.names)<span class="co">#here we replace the extra information by nothing (no characters)</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a>sample.names</span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a>Note that in this case, we have modified the sample names by employing a different way from that used for bacterial dataset. However, the result is exactly the same.</span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-82"><a href="#cb29-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-83"><a href="#cb29-83" aria-hidden="true" tabindex="-1"></a><span class="fu"># 2. Check the quality of the sequencing</span></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a>There are different ways to check the quality of the reads:</span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a><span class="fu">## a) Count the number of reads</span></span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a>It would be nice to check whether we obtained enough reads from the sequencing service. Otherwise, we should ask for the service to repeat the sequencing.</span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)){</span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a>    raw_reads_count <span class="ot">=</span> <span class="fu">rbind</span>(raw_reads_count, </span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a>  } <span class="co">#this loop counts the number of F reads by means of the ShortRead package</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count)<span class="ot">=</span> sample.names <span class="co">#formatting of the output</span></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a>a<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count)</span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a>raw_reads_count2 <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb29-106"><a href="#cb29-106" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnRs)){<span class="co">#do the same with R reads</span></span>
<span id="cb29-107"><a href="#cb29-107" aria-hidden="true" tabindex="-1"></a>  raw_reads_count2 <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count2, </span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnRs[i])))</span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a>} </span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count2)<span class="ot">=</span> sample.names</span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_reads_count2)<span class="ot">=</span> <span class="st">"Number_of_reads"</span></span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a>b<span class="ot">=</span><span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count2),raw_reads_count2)</span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a>a<span class="sc">==</span>b <span class="co">#check that we get the same number of F and R reads</span></span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-116"><a href="#cb29-116" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)],</span>
<span id="cb29-117"><a href="#cb29-117" aria-hidden="true" tabindex="-1"></a>      <span class="fu">min</span>(raw_reads_count))<span class="co">#check which sample accounts for the lowest number of reads. Be careful if you keep the negative control of the sequencing</span></span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)],</span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a>      <span class="fu">max</span>(raw_reads_count))<span class="co">#check which sample accounts for the highest number of reads.</span></span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span><span class="ot">=</span><span class="fu">rownames</span>(raw_reads_count),raw_reads_count),</span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a>            <span class="at">file=</span><span class="st">"Num_raw_reads_ITS.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(raw_reads_count) <span class="co">#please, check the number of reads per sample</span></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a>At this point, we have to check the number of raw reads per sample. The researcher has to verify whether this number is enough or not. It depends on the type of sample, among other parameters.</span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a><span class="fu">## b) Inspect the length of the reads</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a>::: callout-important</span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a>Remember that ITS2 varies in length among different fungi. We have to keep this in mind along the analysis, since we have to skip several steps based on the length of the reads (comparing to the processing of 16S *rRNA* reads).</span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a>In spite of that indicated in the above callout, we will inspect the length of the reads just to confirm the genomics service has performed a good job.</span>
<span id="cb29-135"><a href="#cb29-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-136"><a href="#cb29-136" aria-hidden="true" tabindex="-1"></a>In our case, the genomics service followed a 2x300 bp PE strategy.</span>
<span id="cb29-137"><a href="#cb29-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-141"><a href="#cb29-141" aria-hidden="true" tabindex="-1"></a>reads<span class="ot">=</span>ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co">#save the reads in a new variable with the format of ShortRead package, which is a bit different from standard R variable</span></span>
<span id="cb29-142"><a href="#cb29-142" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">=</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co">#get the length of the reads</span></span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a>counts<span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a>  counts<span class="ot">=</span><span class="fu">rbind</span>(counts,</span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a>               <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width<span class="sc">==</span>uniques[i])))</span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a>}<span class="co">#specific loop to count the number of reads of each length.  </span></span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">=</span>  <span class="fu">cbind</span>(uniques,counts)</span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a><span class="co">#check the histogram</span></span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[,<span class="dv">1</span>],<span class="at">decreasing =</span> <span class="cn">TRUE</span>),]) <span class="co">#Most of the sequences should fall in expected sequence length</span></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a><span class="co">#plotting</span></span>
<span id="cb29-159"><a href="#cb29-159" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main=</span><span class="st">"Forward length distribution"</span>,</span>
<span id="cb29-160"><a href="#cb29-160" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Sequence length"</span>, <span class="at">ylab=</span><span class="st">"Raw reads"</span>)</span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file=</span><span class="st">"Lenght_raw_reads_ITS.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a><span class="fu">## c) Check the quality plots</span></span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a>Check if the overlapping of F and R reads is possible considering the quality of the last nucleotides. Be careful with the quality of R reads (which tends to be worse than that of F reads, especially in the latest part of the reads)</span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-171"><a href="#cb29-171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])<span class="co">#select the specific samples you want to check, in this case 4 and 5</span></span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a><span class="fu"># 3. Filter and trimming step</span></span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-178"><a href="#cb29-178" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb29-179"><a href="#cb29-179" aria-hidden="true" tabindex="-1"></a>If you come from the processing of bacterial reads, you would have notice that here we have skipped the step of FIGARO. That tool needs that all the reads are of the same length, so we cannot use it for fungal (ITS2) dataset.</span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a>filtFs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))<span class="co">#create the "filtered" directory</span></span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a>filtRs<span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a>Since we cannot use FIGARO, we have to test different parameters in the following filtering and trimming step. It is recommended to achieve a compromise between the percentage of retained sequences and their quality. So, run the following command indicating specific values of maxEE according to the quality plots of F and R reads previously visualized in the step 2.c). Write down the results obtained for that maxEE, and then re-run with another maxEE values, until you reach to the best solution.</span>
<span id="cb29-190"><a href="#cb29-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a>out<span class="ot">=</span><span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs, <span class="at">maxN=</span><span class="dv">0</span>, <span class="at">maxEE=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">4</span>), <span class="at">truncQ=</span><span class="dv">2</span>, <span class="at">rm.phix=</span><span class="cn">TRUE</span>, <span class="at">compress=</span><span class="cn">TRUE</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">minLen=</span><span class="dv">50</span>)</span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a><span class="co">#maxN=0 : remove the reads with at least 1 ambiguity</span></span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a><span class="co">#trunQ=2: remove all the reads with one nucleotide with 2 or less quality value</span></span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a><span class="co">#maxEE: maximum expected error for F and R reads</span></span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a><span class="co">#minLen=50: remove reads shorter than 50 bp</span></span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)<span class="co">#important! Check the results</span></span>
<span id="cb29-203"><a href="#cb29-203" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-204"><a href="#cb29-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a>In our case, we selected:</span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a>| Sequencing run | Positions |       | maxEE |       |</span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a>|----------------|-----------|-------|-------|-------|</span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a>|                | **F**     | **R** | **F** | **R** |</span>
<span id="cb29-210"><a href="#cb29-210" aria-hidden="true" tabindex="-1"></a>| **Run5**       |           |       | 3     | 4     |</span>
<span id="cb29-211"><a href="#cb29-211" aria-hidden="true" tabindex="-1"></a>| **Run6**       |           |       | 3     | 4     |</span>
<span id="cb29-212"><a href="#cb29-212" aria-hidden="true" tabindex="-1"></a>| **Run7**       |           |       | 4     | 6     |</span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a>Our reads are partially filtered and trimmed (still have the primers!)</span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a><span class="fu"># 4. Cutadapt: removal of the primers</span></span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a>Our reads still have the primers (which are artificial sequences and can have ambiguities). So, we have to remove them and discard all the sequences in which primers are not found (because if the sequencing run was ok, primers should had been found inside the reads). Lets use cutadapt tool, which does not run in R but in python:</span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">=</span> <span class="st">"GTGARTCATCGAATCTTTG"</span> <span class="co">#sequence of F primer</span></span>
<span id="cb29-224"><a href="#cb29-224" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">=</span> <span class="st">"TCCTCCGCTTATTGATATGC"</span> <span class="co">#sequence of R primer</span></span>
<span id="cb29-225"><a href="#cb29-225" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-229"><a href="#cb29-229" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">=</span> <span class="cf">function</span>(primer) {</span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">=</span><span class="fu">DNAString</span>(primer)  <span class="co">#package Biostrings does not work with characters but with strings, so let's change the class of the primers variables.</span></span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">=</span> <span class="fu">c</span>(<span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), </span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a>              <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna), <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna))</span>
<span id="cb29-235"><a href="#cb29-235" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co">#change from string to character</span></span>
<span id="cb29-236"><a href="#cb29-236" aria-hidden="true" tabindex="-1"></a>} <span class="co">#this function calculates all the possible orientations of the primers</span></span>
<span id="cb29-237"><a href="#cb29-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-238"><a href="#cb29-238" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">=</span> <span class="fu">allOrients</span>(FWD) <span class="co">#pass the function to F and R primers</span></span>
<span id="cb29-239"><a href="#cb29-239" aria-hidden="true" tabindex="-1"></a>REV.orients  <span class="ot">=</span> <span class="fu">allOrients</span>(REV)</span>
<span id="cb29-240"><a href="#cb29-240" aria-hidden="true" tabindex="-1"></a>FWD.orients</span>
<span id="cb29-241"><a href="#cb29-241" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb29-242"><a href="#cb29-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-243"><a href="#cb29-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-246"><a href="#cb29-246" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-247"><a href="#cb29-247" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">=</span><span class="cf">function</span>(primer, fn) {</span>
<span id="cb29-248"><a href="#cb29-248" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">=</span><span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-249"><a href="#cb29-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb29-250"><a href="#cb29-250" aria-hidden="true" tabindex="-1"></a>} <span class="co">#it counts the number of sequence in each orientation</span></span>
<span id="cb29-251"><a href="#cb29-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-252"><a href="#cb29-252" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),<span class="co">#check for instante, in sample number 4</span></span>
<span id="cb29-253"><a href="#cb29-253" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]),</span>
<span id="cb29-254"><a href="#cb29-254" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">4</span>]]),</span>
<span id="cb29-255"><a href="#cb29-255" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">4</span>]]))</span>
<span id="cb29-256"><a href="#cb29-256" aria-hidden="true" tabindex="-1"></a><span class="co">#here we make a summary table. Please, visit DADA2 website for a better interpretation of the table.</span></span>
<span id="cb29-257"><a href="#cb29-257" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-258"><a href="#cb29-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-259"><a href="#cb29-259" aria-hidden="true" tabindex="-1"></a>Now yes, we are running cutadapt, although we have to adapt the reads previously for cutadapt. This tool need some flags that have to be added to the reads. Visit the web of <span class="co">[</span><span class="ot">cutadapt</span><span class="co">](https://cutadapt.readthedocs.io/en/stable/guide.html "User guide of cutadapt")</span>.</span>
<span id="cb29-260"><a href="#cb29-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-263"><a href="#cb29-263" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-264"><a href="#cb29-264" aria-hidden="true" tabindex="-1"></a>cutadapt  <span class="ot">=</span>  <span class="st">"/usr/bin/cutadapt"</span> <span class="co">#path to cutadapt </span></span>
<span id="cb29-265"><a href="#cb29-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-266"><a href="#cb29-266" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb29-267"><a href="#cb29-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-268"><a href="#cb29-268" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">=</span><span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>) <span class="co">#create a directory where processed reads will be saved</span></span>
<span id="cb29-269"><a href="#cb29-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-270"><a href="#cb29-270" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb29-271"><a href="#cb29-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-272"><a href="#cb29-272" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(filtFs))<span class="co">#if you came from the processing of 16S rRNA reads, be careful here. We have to use "filtFs" instead of "fnFs" because we have already performed the filtering and trimming steps.</span></span>
<span id="cb29-273"><a href="#cb29-273" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">=</span><span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(filtRs))<span class="co">#the same comment here</span></span>
<span id="cb29-274"><a href="#cb29-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-275"><a href="#cb29-275" aria-hidden="true" tabindex="-1"></a><span class="co">#Produce arguments for cutadapt</span></span>
<span id="cb29-276"><a href="#cb29-276" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD)  </span>
<span id="cb29-277"><a href="#cb29-277" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">=</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb29-278"><a href="#cb29-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-279"><a href="#cb29-279" aria-hidden="true" tabindex="-1"></a>R1.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>,FWD,<span class="st">"..."</span>, REV.RC) <span class="co">#adding the flags</span></span>
<span id="cb29-280"><a href="#cb29-280" aria-hidden="true" tabindex="-1"></a>R2.flags  <span class="ot">=</span>  <span class="fu">paste0</span>(<span class="st">"-A"</span>,<span class="st">" "</span>,<span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb29-281"><a href="#cb29-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-282"><a href="#cb29-282" aria-hidden="true" tabindex="-1"></a><span class="co">#run cutadapt</span></span>
<span id="cb29-283"><a href="#cb29-283" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb29-284"><a href="#cb29-284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>,<span class="st">"-m"</span>, <span class="dv">1</span>, <span class="st">"--discard-untrimmed"</span>, <span class="st">"-j"</span>,<span class="dv">0</span>, <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], filtFs[i], filtRs[i],<span class="st">"--report=minimal"</span>)) </span>
<span id="cb29-285"><a href="#cb29-285" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-286"><a href="#cb29-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-287"><a href="#cb29-287" aria-hidden="true" tabindex="-1"></a><span class="co">#-n 2: remove the primers</span></span>
<span id="cb29-288"><a href="#cb29-288" aria-hidden="true" tabindex="-1"></a><span class="co">#-m 1: remove empty sequences</span></span>
<span id="cb29-289"><a href="#cb29-289" aria-hidden="true" tabindex="-1"></a><span class="co">#-j 0: automatically detect number of cores</span></span>
<span id="cb29-290"><a href="#cb29-290" aria-hidden="true" tabindex="-1"></a><span class="co">#-0: output files</span></span>
<span id="cb29-291"><a href="#cb29-291" aria-hidden="true" tabindex="-1"></a><span class="co">#-i: input files. We use the filtered and trimmed reads</span></span>
<span id="cb29-292"><a href="#cb29-292" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-293"><a href="#cb29-293" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-294"><a href="#cb29-294" aria-hidden="true" tabindex="-1"></a>Now, we check whether cutadapt has removed all the primers. For that purpose, we pass the previously created function to check the number of times each primer appears in our dataset:</span>
<span id="cb29-295"><a href="#cb29-295" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-298"><a href="#cb29-298" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-299"><a href="#cb29-299" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(<span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),<span class="co">#Here we can indicate the number of the sample we want to check. In this case, sample number 4</span></span>
<span id="cb29-300"><a href="#cb29-300" aria-hidden="true" tabindex="-1"></a>      <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb29-301"><a href="#cb29-301" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">4</span>]]),</span>
<span id="cb29-302"><a href="#cb29-302" aria-hidden="true" tabindex="-1"></a>      <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">4</span>]]))</span>
<span id="cb29-303"><a href="#cb29-303" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-304"><a href="#cb29-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-305"><a href="#cb29-305" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb29-306"><a href="#cb29-306" aria-hidden="true" tabindex="-1"></a>Note that in some cases, after the removal of the primers by cutadapt, the function primerHits still finds some primers in our dataset. Do no pannic; these differences are normal because cutadapt and that DADA2 functions are based on different algorithms.</span>
<span id="cb29-307"><a href="#cb29-307" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-308"><a href="#cb29-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-309"><a href="#cb29-309" aria-hidden="true" tabindex="-1"></a><span class="fu"># 5. DADA2: machine learning</span></span>
<span id="cb29-310"><a href="#cb29-310" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-311"><a href="#cb29-311" aria-hidden="true" tabindex="-1"></a>We are going to use DADA2 to denoise, infer the samples and to correct the expected errors by applying different functions of the package</span>
<span id="cb29-312"><a href="#cb29-312" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-315"><a href="#cb29-315" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-316"><a href="#cb29-316" aria-hidden="true" tabindex="-1"></a><span class="co"># First, learn error rates</span></span>
<span id="cb29-317"><a href="#cb29-317" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">=</span><span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb29-318"><a href="#cb29-318" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">=</span><span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread=</span>T, <span class="at">verbose=</span><span class="dv">1</span>)</span>
<span id="cb29-319"><a href="#cb29-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-320"><a href="#cb29-320" aria-hidden="true" tabindex="-1"></a><span class="co"># Lets plot the errors in each position</span></span>
<span id="cb29-321"><a href="#cb29-321" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-322"><a href="#cb29-322" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-323"><a href="#cb29-323" aria-hidden="true" tabindex="-1"></a><span class="co">#these plots show the possible errors made for each possible mutation (A&gt;C, A&gt;G, etc). Dots should be close to the red line.</span></span>
<span id="cb29-324"><a href="#cb29-324" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-325"><a href="#cb29-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-326"><a href="#cb29-326" aria-hidden="true" tabindex="-1"></a><span class="fu"># 6. DADA2: Sample inference</span></span>
<span id="cb29-327"><a href="#cb29-327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-330"><a href="#cb29-330" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-331"><a href="#cb29-331" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">=</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err=</span>errF, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-332"><a href="#cb29-332" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">=</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err=</span>errR, <span class="at">multithread=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-333"><a href="#cb29-333" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">4</span>]]<span class="co">#check the inferred fourth sample (it's just an example). We will obtain the number of true sequence variants from the X number of unique sequences.</span></span>
<span id="cb29-334"><a href="#cb29-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-335"><a href="#cb29-335" aria-hidden="true" tabindex="-1"></a><span class="co"># Set sample names</span></span>
<span id="cb29-336"><a href="#cb29-336" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">=</span> sample.names</span>
<span id="cb29-337"><a href="#cb29-337" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">=</span> sample.names</span>
<span id="cb29-338"><a href="#cb29-338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-339"><a href="#cb29-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-340"><a href="#cb29-340" aria-hidden="true" tabindex="-1"></a><span class="fu"># 7. Merge sequences</span></span>
<span id="cb29-341"><a href="#cb29-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-342"><a href="#cb29-342" aria-hidden="true" tabindex="-1"></a>In this step we are going to overlap F and R reads coming from the sample sample.</span>
<span id="cb29-343"><a href="#cb29-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-346"><a href="#cb29-346" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-347"><a href="#cb29-347" aria-hidden="true" tabindex="-1"></a>mergers<span class="ot">=</span><span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose=</span><span class="cn">TRUE</span>)</span>
<span id="cb29-348"><a href="#cb29-348" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">4</span>]]) <span class="co">#check the output for sample number 4</span></span>
<span id="cb29-349"><a href="#cb29-349" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-350"><a href="#cb29-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-351"><a href="#cb29-351" aria-hidden="true" tabindex="-1"></a>Now, we are going to construct an amplicon sequence variant table (ASV table).</span>
<span id="cb29-352"><a href="#cb29-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-353"><a href="#cb29-353" aria-hidden="true" tabindex="-1"></a>Be careful, because we will get one ASV table per sequencing run. Thus, as we got 3 runs for fungal amplicons, we will get 3 different ASV table.</span>
<span id="cb29-354"><a href="#cb29-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-357"><a href="#cb29-357" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-358"><a href="#cb29-358" aria-hidden="true" tabindex="-1"></a>seqtab_run5<span class="ot">=</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb29-359"><a href="#cb29-359" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_run5) <span class="co">#indicates the number of samples (including negative controls) and the number of ASVs</span></span>
<span id="cb29-360"><a href="#cb29-360" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_run5, <span class="st">"~/Platano_PLEC/PLEC_muestreo2022/310523_ITS/reads/seqtab_run5.rds"</span>) <span class="co">#save the seqtab in .rds format</span></span>
<span id="cb29-361"><a href="#cb29-361" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-362"><a href="#cb29-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-363"><a href="#cb29-363" aria-hidden="true" tabindex="-1"></a><span class="fu"># 8. Merge sequence tables from different runs</span></span>
<span id="cb29-364"><a href="#cb29-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-365"><a href="#cb29-365" aria-hidden="true" tabindex="-1"></a>As indicated before, in this step we are going to merge the sequence tables coming from different runs. So, firstly, we have to load all of them to the current project. Please, note that all the seqtabs have to be in the same working directory!</span>
<span id="cb29-366"><a href="#cb29-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-369"><a href="#cb29-369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-370"><a href="#cb29-370" aria-hidden="true" tabindex="-1"></a>seqtab_run5<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run5.rds"</span>)<span class="co">#load all the .rds files in the same directory</span></span>
<span id="cb29-371"><a href="#cb29-371" aria-hidden="true" tabindex="-1"></a>seqtab_run6<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run6rds"</span>)</span>
<span id="cb29-372"><a href="#cb29-372" aria-hidden="true" tabindex="-1"></a>seqtab_run7<span class="ot">=</span><span class="fu">readRDS</span>(<span class="st">"seqtab_run7rds"</span>)</span>
<span id="cb29-373"><a href="#cb29-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-374"><a href="#cb29-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-375"><a href="#cb29-375" aria-hidden="true" tabindex="-1"></a>mergedSeqTab<span class="ot">=</span><span class="fu">mergeSequenceTables</span>(seqtab_run5,seqtab_run6,seqtab_run7,<span class="at">repeats=</span><span class="st">"sum"</span>)</span>
<span id="cb29-376"><a href="#cb29-376" aria-hidden="true" tabindex="-1"></a><span class="co">#with "sum" we indicate that we want to join and sum all the reads that are in samples with the same name. Why do we do that? </span></span>
<span id="cb29-377"><a href="#cb29-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-378"><a href="#cb29-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-379"><a href="#cb29-379" aria-hidden="true" tabindex="-1"></a><span class="fu"># 9. Chimeral removal</span></span>
<span id="cb29-380"><a href="#cb29-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-381"><a href="#cb29-381" aria-hidden="true" tabindex="-1"></a>As the sequencing process is based on PCRs, chimeric sequences are expect to appear. So, let's remove them</span>
<span id="cb29-382"><a href="#cb29-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-385"><a href="#cb29-385" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-386"><a href="#cb29-386" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">=</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method=</span><span class="st">"consensus"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>, <span class="at">verbose=</span><span class="cn">TRUE</span>)<span class="co">#if you just are working on one dataset (coming from just one sequencing run, instead of "mergedSeqTab", you have to write "seqt_tab" or the name of your seqtab)</span></span>
<span id="cb29-387"><a href="#cb29-387" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co">#indicates the number of samples and ASVs (but not the number of sequences)</span></span>
<span id="cb29-388"><a href="#cb29-388" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-389"><a href="#cb29-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-390"><a href="#cb29-390" aria-hidden="true" tabindex="-1"></a><span class="fu"># 10. Other steps: flter the ASVs by their length</span></span>
<span id="cb29-391"><a href="#cb29-391" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-392"><a href="#cb29-392" aria-hidden="true" tabindex="-1"></a>Let's check the number of ASVs and their length</span>
<span id="cb29-393"><a href="#cb29-393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-396"><a href="#cb29-396" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-397"><a href="#cb29-397" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim)))  <span class="co">#Number of ASV of each length</span></span>
<span id="cb29-398"><a href="#cb29-398" aria-hidden="true" tabindex="-1"></a>  <span class="co">#first line: length of the sequences</span></span>
<span id="cb29-399"><a href="#cb29-399" aria-hidden="true" tabindex="-1"></a>  <span class="co">#second line: number of sequences of each length</span></span>
<span id="cb29-400"><a href="#cb29-400" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-401"><a href="#cb29-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-402"><a href="#cb29-402" aria-hidden="true" tabindex="-1"></a>Calculate the number of sequences of each ASV that that each length:</span>
<span id="cb29-403"><a href="#cb29-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-406"><a href="#cb29-406" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-407"><a href="#cb29-407" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">=</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb29-408"><a href="#cb29-408" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb29-409"><a href="#cb29-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-410"><a href="#cb29-410" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">length=</span><span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count=</span>reads.per.seqlen)</span>
<span id="cb29-411"><a href="#cb29-411" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x=</span>length, <span class="at">y=</span>count)) <span class="sc">+</span> <span class="fu">geom_col</span>()</span>
<span id="cb29-412"><a href="#cb29-412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-413"><a href="#cb29-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-414"><a href="#cb29-414" aria-hidden="true" tabindex="-1"></a><span class="fu"># 11. Taxonomic classification</span></span>
<span id="cb29-415"><a href="#cb29-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-416"><a href="#cb29-416" aria-hidden="true" tabindex="-1"></a>We are going to classify our ITS2 amplicons against <span class="co">[</span><span class="ot">UNITE</span><span class="co">](https://doi.org/10.15156/BIO/2938067 "Click to the paper")</span> v.9.0 databse</span>
<span id="cb29-417"><a href="#cb29-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-420"><a href="#cb29-420" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-421"><a href="#cb29-421" aria-hidden="true" tabindex="-1"></a>taxa_unite <span class="ot">=</span><span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/home/databases/sh_general_release_dynamic_25.07.2023.fasta"</span>, <span class="at">multithread=</span><span class="cn">TRUE</span>)<span class="co">#indicate the path were your database is located</span></span>
<span id="cb29-422"><a href="#cb29-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-423"><a href="#cb29-423" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">=</span> seqtab.nochim <span class="co">#edit the tables</span></span>
<span id="cb29-424"><a href="#cb29-424" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">=</span> <span class="fu">t</span>(ASV)</span>
<span id="cb29-425"><a href="#cb29-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-426"><a href="#cb29-426" aria-hidden="true" tabindex="-1"></a>taxa_na <span class="ot">=</span> <span class="fu">apply</span>(taxa_unite,<span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[,<span class="sc">-</span><span class="dv">7</span>] <span class="co">#replace the "NA" values by "unclassified" in those case in which an ASV have not been identified at a specific taxonomic rank</span></span>
<span id="cb29-427"><a href="#cb29-427" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na<span class="ot">=</span>taxa_rdp_na[,<span class="sc">-</span>(<span class="dv">7</span><span class="sc">:</span><span class="dv">15</span>)]</span>
<span id="cb29-428"><a href="#cb29-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-429"><a href="#cb29-429" aria-hidden="true" tabindex="-1"></a><span class="co">#we are going to modify a bit the tables so that the name of ASVs looks like "ASV00001" (in case we have 10000 ASVs or more). In case we have just 100 ASVs, they will have that code: "ASV001"</span></span>
<span id="cb29-430"><a href="#cb29-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-431"><a href="#cb29-431" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">=</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb29-432"><a href="#cb29-432" aria-hidden="true" tabindex="-1"></a>names <span class="ot">=</span><span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co">#As many 0 as digits</span></span>
<span id="cb29-433"><a href="#cb29-433" aria-hidden="true" tabindex="-1"></a>ASV_names<span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb29-434"><a href="#cb29-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-435"><a href="#cb29-435" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_na,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),<span class="fu">as.data.frame</span>(ASVt,<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb29-436"><a href="#cb29-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-437"><a href="#cb29-437" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">=</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb29-438"><a href="#cb29-438" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb29-439"><a href="#cb29-439" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">=</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span>
<span id="cb29-440"><a href="#cb29-440" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-441"><a href="#cb29-441" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_table_classified_raw, <span class="at">file=</span><span class="st">"ASV_table_classified_raw_wirhMockwithPlastids.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb29-442"><a href="#cb29-442" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-443"><a href="#cb29-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-444"><a href="#cb29-444" aria-hidden="true" tabindex="-1"></a>Now, we have our ASV table with the corresponding taxonomic classification. But, it still potentially includes some sequences that could be erroneous.</span>
<span id="cb29-445"><a href="#cb29-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-446"><a href="#cb29-446" aria-hidden="true" tabindex="-1"></a><span class="fu"># 12. MOCK Community: setting the sequencing detection limit</span></span>
<span id="cb29-447"><a href="#cb29-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-448"><a href="#cb29-448" aria-hidden="true" tabindex="-1"></a>The Mock Community we used in our sequencing runs just include sequences of two fungi. Thus, we will consider that the detection limit previously established for bacterial dataset is the same as for fungi. Ours was 0.001207% of the sequences. Let's use it:</span>
<span id="cb29-449"><a href="#cb29-449" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-452"><a href="#cb29-452" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-453"><a href="#cb29-453" aria-hidden="true" tabindex="-1"></a>ASV_sums<span class="ot">=</span><span class="fu">rowSums</span>(ASV_table_classified_raw[,<span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(ASV_table_classified_raw)])</span>
<span id="cb29-454"><a href="#cb29-454" aria-hidden="true" tabindex="-1"></a>sum.total<span class="ot">=</span><span class="fu">sum</span>(ASV_sums)<span class="co"># Get the total number of sequences</span></span>
<span id="cb29-455"><a href="#cb29-455" aria-hidden="true" tabindex="-1"></a>nseq_cutoff<span class="ot">=</span>(<span class="fl">0.001207</span><span class="sc">/</span><span class="dv">100</span>)<span class="sc">*</span>sum.total<span class="co"># Apply the percentage to sequence number</span></span>
<span id="cb29-456"><a href="#cb29-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-457"><a href="#cb29-457" aria-hidden="true" tabindex="-1"></a><span class="co"># Now, filter the table accordingly:</span></span>
<span id="cb29-458"><a href="#cb29-458" aria-hidden="true" tabindex="-1"></a>ASV_filtered<span class="ot">=</span>ASV_table_classified_raw[<span class="fu">which</span>(ASV_sums<span class="sc">&gt;</span>nseq_cutoff),] <span class="co">#we are retaining just those ASVs accounting for more sequences that that determined by the cut-off</span></span>
<span id="cb29-459"><a href="#cb29-459" aria-hidden="true" tabindex="-1"></a>ASV_table<span class="ot">=</span>ASV_filtered[<span class="fu">order</span>(ASV_filtered[[<span class="st">"ASV_names"</span>]]),]<span class="co"># Sort table in ascending order of ASV names</span></span>
<span id="cb29-460"><a href="#cb29-460" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-461"><a href="#cb29-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-462"><a href="#cb29-462" aria-hidden="true" tabindex="-1"></a><span class="fu"># 13. Removal of erronous sequences and taxonomy refinement</span></span>
<span id="cb29-463"><a href="#cb29-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-464"><a href="#cb29-464" aria-hidden="true" tabindex="-1"></a>Let's remove all the undesired sequences (if we are working with plant endophytes, we will have retained plant sequences). We will also remove those ASVs not classified even at kingdom level.</span>
<span id="cb29-465"><a href="#cb29-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-468"><a href="#cb29-468" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-469"><a href="#cb29-469" aria-hidden="true" tabindex="-1"></a>ASV_final<span class="ot">=</span>ASV_table[(<span class="fu">which</span>(ASV_table<span class="sc">$</span>Kingdom<span class="sc">!=</span><span class="st">"unclassified"</span>)),] <span class="co">#remove all the sequences not classified at Kingdom level</span></span>
<span id="cb29-470"><a href="#cb29-470" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-471"><a href="#cb29-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-472"><a href="#cb29-472" aria-hidden="true" tabindex="-1"></a>Sometimes, when we have worked with plant tissues, we can find some fungi that are adscribed to "*incertae sedis*" group (missclassified) even at phylum level. They tend to be sequences of the plant host. So, let's verify it and remove then in case we find these sequences:</span>
<span id="cb29-473"><a href="#cb29-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-476"><a href="#cb29-476" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-477"><a href="#cb29-477" aria-hidden="true" tabindex="-1"></a>incertae_phy<span class="ot">=</span>ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"p__Fungi_phy_Incertae_sedis"</span>),] <span class="co">#UNITE database writes "p__" prior to the name of the phyla</span></span>
<span id="cb29-478"><a href="#cb29-478" aria-hidden="true" tabindex="-1"></a>seq_incertae<span class="ot">=</span><span class="fu">as.list</span>(incertae_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb29-479"><a href="#cb29-479" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_incertae, <span class="at">names=</span>incertae_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"incertaesedis_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)<span class="co">#save the fasta file of these ASVs</span></span>
<span id="cb29-480"><a href="#cb29-480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-481"><a href="#cb29-481" aria-hidden="true" tabindex="-1"></a><span class="co">#Let's perform a comparison of these sequences against the NCBI GenBank database by BLASTn</span></span>
<span id="cb29-482"><a href="#cb29-482" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query incertaesedis_phylum.fas -db /home/databases/nt -out incertae_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-483"><a href="#cb29-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-484"><a href="#cb29-484" aria-hidden="true" tabindex="-1"></a><span class="co">#The output of the BLAST ("incertae_phylum_hits.txt") have to be manually checked.</span></span>
<span id="cb29-485"><a href="#cb29-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-486"><a href="#cb29-486" aria-hidden="true" tabindex="-1"></a><span class="co">#In our case, all the sequences are related to *Musa* spp. and other eukaryots that are not fungi, so we are going to remove all these sequences:</span></span>
<span id="cb29-487"><a href="#cb29-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-488"><a href="#cb29-488" aria-hidden="true" tabindex="-1"></a>ASV_final2 <span class="ot">=</span> ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum<span class="sc">!=</span><span class="st">"p__Fungi_phy_Incertae_sedis"</span>)),]</span>
<span id="cb29-489"><a href="#cb29-489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-490"><a href="#cb29-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-491"><a href="#cb29-491" aria-hidden="true" tabindex="-1"></a>Repeat the same with the unclassified sequences at phylum level:</span>
<span id="cb29-492"><a href="#cb29-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-495"><a href="#cb29-495" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-496"><a href="#cb29-496" aria-hidden="true" tabindex="-1"></a>unclassified_phy<span class="ot">=</span>ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum<span class="sc">==</span><span class="st">"unclassified"</span>),]</span>
<span id="cb29-497"><a href="#cb29-497" aria-hidden="true" tabindex="-1"></a>seq_unclassified<span class="ot">=</span> <span class="fu">as.list</span>(unclassified_phy<span class="sc">$</span>ASV_seqs)</span>
<span id="cb29-498"><a href="#cb29-498" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclassified, <span class="at">names=</span>unclassified_phy<span class="sc">$</span>ASV_names, <span class="at">file.out=</span><span class="st">"unclassified_phylum.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span><span class="dv">1000</span> , <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-499"><a href="#cb29-499" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programas/ncbi-blast-2.13.0+/bin/blastn -query unclassified_phylum.fas -db /home/databases/nt -out unclassified_phylum_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 5 -parse_deflines -num_threads 10"</span>),<span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb29-500"><a href="#cb29-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-501"><a href="#cb29-501" aria-hidden="true" tabindex="-1"></a><span class="co">#We found a lot of missclassified ASVs at phylum, so we are going to remove them especifically:</span></span>
<span id="cb29-502"><a href="#cb29-502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-503"><a href="#cb29-503" aria-hidden="true" tabindex="-1"></a>ASV_final3 <span class="ot">=</span> ASV_final2[(<span class="fu">which</span>(ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0004"</span><span class="sc">&amp;</span></span>
<span id="cb29-504"><a href="#cb29-504" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0012"</span><span class="sc">&amp;</span></span>
<span id="cb29-505"><a href="#cb29-505" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0109"</span><span class="sc">&amp;</span></span>
<span id="cb29-506"><a href="#cb29-506" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0140"</span><span class="sc">&amp;</span></span>
<span id="cb29-507"><a href="#cb29-507" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0269"</span><span class="sc">&amp;</span></span>
<span id="cb29-508"><a href="#cb29-508" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0627"</span><span class="sc">&amp;</span></span>
<span id="cb29-509"><a href="#cb29-509" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0654"</span><span class="sc">&amp;</span></span>
<span id="cb29-510"><a href="#cb29-510" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1112"</span><span class="sc">&amp;</span></span>
<span id="cb29-511"><a href="#cb29-511" aria-hidden="true" tabindex="-1"></a>                                ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0938"</span><span class="sc">&amp;</span></span>
<span id="cb29-512"><a href="#cb29-512" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0911"</span><span class="sc">&amp;</span></span>
<span id="cb29-513"><a href="#cb29-513" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0731"</span><span class="sc">&amp;</span></span>
<span id="cb29-514"><a href="#cb29-514" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0844"</span><span class="sc">&amp;</span></span>
<span id="cb29-515"><a href="#cb29-515" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV0923"</span><span class="sc">&amp;</span></span>
<span id="cb29-516"><a href="#cb29-516" aria-hidden="true" tabindex="-1"></a>                                  ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1080"</span><span class="sc">&amp;</span></span>
<span id="cb29-517"><a href="#cb29-517" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1096"</span><span class="sc">&amp;</span></span>
<span id="cb29-518"><a href="#cb29-518" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1147"</span><span class="sc">&amp;</span></span>
<span id="cb29-519"><a href="#cb29-519" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1291"</span><span class="sc">&amp;</span></span>
<span id="cb29-520"><a href="#cb29-520" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1321"</span><span class="sc">&amp;</span></span>
<span id="cb29-521"><a href="#cb29-521" aria-hidden="true" tabindex="-1"></a>                                 ASV_final2<span class="sc">$</span>ASV_names<span class="sc">!=</span><span class="st">"ASV1335"</span>)),]</span>
<span id="cb29-522"><a href="#cb29-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-523"><a href="#cb29-523" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-524"><a href="#cb29-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-525"><a href="#cb29-525" aria-hidden="true" tabindex="-1"></a>Eventually, save the definitive ASV table:</span>
<span id="cb29-526"><a href="#cb29-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-529"><a href="#cb29-529" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-530"><a href="#cb29-530" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">" "</span><span class="ot">=</span><span class="fu">rownames</span>(ASV_final3),ASV_final3),<span class="at">file=</span><span class="st">"ASV_Hongos_FINAL.txt"</span>, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>,<span class="at">row.names =</span>F)</span>
<span id="cb29-531"><a href="#cb29-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>