<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Banana - The microbiota of banana plants</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Banana</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" rel="" target="" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-processing-of-miseq-reads" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Processing of MiSeq reads</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-processing-of-miseq-reads">    
        <li>
    <a class="dropdown-item" href="./Bacteria_reads_processing.html" rel="" target="">
 <span class="dropdown-text">16S rRNA reads</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_reads_processing.html" rel="" target="">
 <span class="dropdown-text">ITS2 reads</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-ecological-analyses" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Ecological analyses</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-ecological-analyses">    
        <li>
    <a class="dropdown-item" href="./Bacteria_data_analysis.html" rel="" target="">
 <span class="dropdown-text">Bacterial dataset</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./Fungi_data_analysis.html" rel="" target="">
 <span class="dropdown-text">Fungal dataset</span></a>
  </li>  
    </ul>
  </li>
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/anitalasa" rel="" target=""><i class="bi bi-github" role="img" aria-label="GitHub">
</i> 
 <span class="menu-text"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://orcid.org/0000-0003-3783-7157" rel="" target="">
 <span class="menu-text"><img src="images/orcid.png" title="ORCID" alt="ORCID" style="height:20px;"></span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://grupos.eez.csic.es/mae/" rel="" target="">
 <span class="menu-text"><img src="images/internet.png" title="web research group" alt="web research group" style="height:20px;"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#scripts-to-process-and-analyze-high-throughput-sequencing-reads" id="toc-scripts-to-process-and-analyze-high-throughput-sequencing-reads" class="nav-link active" data-scroll-target="#scripts-to-process-and-analyze-high-throughput-sequencing-reads">Scripts to process and analyze high-throughput sequencing reads</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">1. Introduction</a>
  <ul class="collapse">
  <li><a href="#but..-how-do-banana-plants-develop" id="toc-but..-how-do-banana-plants-develop" class="nav-link" data-scroll-target="#but..-how-do-banana-plants-develop">1.1 But.. how do banana plants develop?</a></li>
  </ul></li>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link" data-scroll-target="#experimental-design">2. Experimental design</a></li>
  <li><a href="#sequencing-scheme" id="toc-sequencing-scheme" class="nav-link" data-scroll-target="#sequencing-scheme">3. Sequencing scheme</a></li>
  <li><a href="#sample-naming" id="toc-sample-naming" class="nav-link" data-scroll-target="#sample-naming">4. Sample naming</a></li>
  <li><a href="#technical-issues" id="toc-technical-issues" class="nav-link" data-scroll-target="#technical-issues">5. Technical issues</a></li>
  <li><a href="#what-will-i-find-in-this-github-project" id="toc-what-will-i-find-in-this-github-project" class="nav-link" data-scroll-target="#what-will-i-find-in-this-github-project">6. What will I find in this GitHub project?</a>
  <ul class="collapse">
  <li><a href="#scripts-for-the-processing-of-miseq-reads" id="toc-scripts-for-the-processing-of-miseq-reads" class="nav-link" data-scroll-target="#scripts-for-the-processing-of-miseq-reads">6.1 Scripts for the processing of MiSeq reads</a></li>
  <li><a href="#script-for-the-ecological-analyses" id="toc-script-for-the-ecological-analyses" class="nav-link" data-scroll-target="#script-for-the-ecological-analyses">6.2 Script for the ecological analyses</a></li>
  </ul></li>
  <li><a href="#authors" id="toc-authors" class="nav-link" data-scroll-target="#authors">7. Authors</a></li>
  <li><a href="#how-to-cite" id="toc-how-to-cite" class="nav-link" data-scroll-target="#how-to-cite">8. How to cite</a>
  <ul class="collapse">
  <li><a href="#main-paper" id="toc-main-paper" class="nav-link" data-scroll-target="#main-paper">8.1 Main paper</a></li>
  <li><a href="#repository" id="toc-repository" class="nav-link" data-scroll-target="#repository">8.2 Repository</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">9. References</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">The microbiota of banana plants</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="scripts-to-process-and-analyze-high-throughput-sequencing-reads" class="level2">
<h2 class="anchored" data-anchor-id="scripts-to-process-and-analyze-high-throughput-sequencing-reads">Scripts to process and analyze high-throughput sequencing reads</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Dibujo.jpg" class="img-fluid figure-img" width="500"></p>
<figcaption class="figure-caption">Figure 1. Graphic representation of banana plants</figcaption>
</figure>
</div>
<p>This repository includes all the scripts employed for the processing of amplicon sequencing reads and the scripts needed for all the subsequent ecological analyses.</p>
</section>
<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">1. Introduction</h2>
<p>These analyses were made for the project <strong>EVOMICROBIA</strong> (<em>Evolución dirigida de consorcios microbianos mejorados para el biocontrol de la Fusariosis vascular del Plátano de Canarias</em>). The main aim of this work was to analyze the microbial communities (bacteria and fungi) associated to the roots of banana plants (<em>Musa acuminata</em> cv. Pequeña Enana synonymous with Dwarf Cavendish). We also aimed at determining whether microbial members are transferred from mother to sucker plants through the root system, and if possible, to decipher which part of the microbiota is transferred from mothers to first suckers and from these to second sucker plants. We focused on two plots(orchards) located in different points in Tenerife Island (Canary Islands, Spain), with different agronomic management. In the following map, the location of both plots its indicated (Figure 2):</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Mapa.jpg" class="img-fluid figure-img" width="500"></p>
<figcaption class="figure-caption">Figure 2. Map of Tenerife Island (Canary Islands, Spain) and photographs of both orchards under study</figcaption>
</figure>
</div>
<section id="but..-how-do-banana-plants-develop" class="level3">
<h3 class="anchored" data-anchor-id="but..-how-do-banana-plants-develop">1.1 But.. how do banana plants develop?</h3>
<p>Banana mats are comprised of the underground rhizome (also named <em>corm</em>) from which “suckers” or new clonal shoots arise close to mother plants, the pseudostem (composed by tightly packed overlapping leaf sheaths), stem and leafs. Thus, suckers are forms of vegetative reproduction, and can develop and produce an inflorescence when they reach the adult state. Once mother plants produce fruit bunches and they are harvested, mother plants are cut since they do not produce bananas any more. This cycle repeats periodically, so suckers become mothers and so happens with all the clonal progeny (clonal “offspring”).</p>
</section>
</section>
<section id="experimental-design" class="level2">
<h2 class="anchored" data-anchor-id="experimental-design">2. Experimental design</h2>
<p>Samples were taken in two consecutive campaigns:</p>
<ol type="1">
<li><p><strong>Year 2022</strong>: 12 healthy banana mother plants whose sucker plants were actively growing were selected from each of the orchards. Samples from rhizosphere soil of the 12 mother plants were manually collected, along with rhizosphere soils from the sucker plant growing close to the each mother. Roots from each mother and sucker plants were also collected in each orchard Thus, a total of 12 replicates x 2 plants in different developmental state (mother and sucker) x 2 orchards x 2 plant compartments = 96 samples.</p></li>
<li><p><strong>Year 2023</strong>: the same sampling scheme was followed. However, it should be taken into account that plants that were suckers in 2022 acted as mothers in 2023, and their suckers were considered as the second suckers.</p></li>
</ol>
<p>Have a look at the next table that summarizes the sampling scheme.</p>
<table class="table">
<caption><em>Number of samples analyzed in each year, plant compartment and plot</em>.</caption>
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
</colgroup>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong>Year 2022</strong></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="even">
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Plot 1 (South)</strong></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Plot 2 (North)</strong></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Mother</strong></td>
<td style="text-align: center;"><strong>Sucker</strong></td>
<td style="text-align: center;"><strong>Mother</strong></td>
<td style="text-align: center;"><strong>Sucker</strong></td>
</tr>
<tr class="even">
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Rhizosphere</strong></td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
</tr>
<tr class="odd">
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Endosphere</strong></td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Year 2023</strong></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
<td style="text-align: center;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Rhizosphere</strong></td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
</tr>
<tr class="even">
<td style="text-align: center;"></td>
<td style="text-align: center;"><strong>Endosphere</strong></td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
<td style="text-align: center;">12</td>
</tr>
</tbody>
</table>
</section>
<section id="sequencing-scheme" class="level2">
<h2 class="anchored" data-anchor-id="sequencing-scheme">3. Sequencing scheme</h2>
<p>We started from 192 samples, from which total DNA was isolated (in the case of the roots, they were surface sterilized previously). Thus, we got 192 DNA samples. However, it should be taken into account that we wanted to analyze both bacterial and fungal communities. Thus, we had to sequence 192 x 2 = 384 samples (and also the controls). Due to the high amount of samples and the different sampling campaigns, our sequencing plan was divided into different runs. Furthermore, the sequencing service (<a href="https://www.ipb.csic.es/servicios/Genomica/index.html" title="Click to Genomics Service"><em>Instituto de Parasitología y Biomedicina López-Neyra</em></a>, CSIC, Granada) had to resequence some samples due to technical issues. Thus, our sequencing project had to be divided into the following sequencing runs:</p>
<ul>
<li><p>Bacteria:</p>
<ul>
<li><p>Run 1: most of the samples corresponding to the first sampling campaign (2022).</p></li>
<li><p>Run 2: some additional samples corresponding to year 2022.</p></li>
<li><p>Run 3: samples corresponding to the second campaign (2023).</p></li>
<li><p>Run 4: resequencing of some samples corresponding to year 2023.</p></li>
</ul></li>
</ul>
<!-- -->
<ul>
<li><p>Fungi:</p>
<ul>
<li><p>Run 5: most of the samples corresponding to the first campaign (2022).</p></li>
<li><p>Run 6: other samples corresponding to 2022.</p></li>
<li><p>Run 7: all the samples corresponding to 2023</p></li>
</ul></li>
</ul>
<p>It should be pointed out that in each sequencing run (just in the case of the bacterial dataset) three replicates of the mock community ZymoBIOMICS Microbial Community Standard II (ZYMO Research) were included.</p>
</section>
<section id="sample-naming" class="level2">
<h2 class="anchored" data-anchor-id="sample-naming">4. Sample naming</h2>
<p>The name of the samples was composed by a combination of alphanumeric characters.</p>
<ul>
<li><p>First character: letter “A” for samples obtained in 2022, and “B” for those obtained in 2023.</p></li>
<li><p>Second character: numer “1” for the plot located in the south orchard, and “2” for the plot in the north of the island.</p></li>
<li><p>Third character: letter “M” for mother plants, and “S” for suckers.</p></li>
<li><p>Fourth and fifth character: the number of the replicate. It should be taken into account that from 1 to 9, replicates are names as 01-09.</p></li>
<li><p>Sixth character: letter “E” for root endosphere and “R” for rhizosphere soil samples.</p></li>
</ul>
<p>Thus, rhizosphere soil of the mother plant number 7 located in the north orchard sampled in 2023 will be named as B2M07E.</p>
</section>
<section id="technical-issues" class="level2">
<h2 class="anchored" data-anchor-id="technical-issues">5. Technical issues</h2>
<p>-In the case of <strong>bacteria</strong>, the hypervariable regions V3-V4 of the prokaryotic gene <em>16S rRNA</em> were sequenced. For that purpose, the primer pair Pro341F and Pro805R was employed (<a href="https://doi.org/10.1371/journal.pone.0105592" title="Go to the paper">Takahashi et al., 2014</a>)</p>
<p>-In the case of <strong>fungi</strong>, the ITS2 region was sequenced. For that purpose, the primer pair ITS4 and fiTS7 was employed (White et al., 1990 and <a href="https://doi.org/10.1111/j.1574-6941.2012.01437.x" title="Go to the paper">Ihrmark et al., 2012</a></p>
<p>-In order to diminish the amplification and sequencing of the DNA from the plant host (from mitochondria, chloroplast etc), DNA was treated with the PNA PCR Clamps proposed by <a href="https://doi.org/10.1038/nmeth.2634" title="Go to the paper">Lunderberg et al., (2013)</a></p>
<p>-MiSeq Illumina platform was used for the sequencing, and two types of Paired-End strategies were followed:</p>
<ul>
<li><p><strong>Runs 1, 2, 3, 5, 6 and 7</strong>: 2x275 PE strategy</p></li>
<li><p><strong>Run 4</strong>: 2x300 PE strategy</p></li>
</ul>
<p>The remaining technical issues are explained in the Research Article</p>
</section>
<section id="what-will-i-find-in-this-github-project" class="level2">
<h2 class="anchored" data-anchor-id="what-will-i-find-in-this-github-project">6. What will I find in this GitHub project?</h2>
<p>In this repository, you will find all the scripts that we used for two types of approaches: 1) Processing of Illumina MiSeq reads and 2) Ecological analyses</p>
<section id="scripts-for-the-processing-of-miseq-reads" class="level3">
<h3 class="anchored" data-anchor-id="scripts-for-the-processing-of-miseq-reads">6.1 Scripts for the processing of MiSeq reads</h3>
<p>These scripts explain all the steps to process the Illumina reads, from the fastq files to a filtered ASV table.</p>
<p><a href="./Bacteria_reads_processing.html">16S rRNA reads</a></p>
<p><a href="./Fungi_reads_processing.html">ITS2 reads</a></p>
</section>
<section id="script-for-the-ecological-analyses" class="level3">
<h3 class="anchored" data-anchor-id="script-for-the-ecological-analyses">6.2 Script for the ecological analyses</h3>
<p>All the steps from the previously obtained ASV table to the alpha and beta diversity, differential abundance analysis, etc.</p>
<p><a href="./Bacteria_data_analysis.html">Analysis of bacterial dataset</a></p>
<p><a href="./Fungi_data_analysis.html">Analysis of fungal dataset</a></p>
</section>
</section>
<section id="authors" class="level2">
<h2 class="anchored" data-anchor-id="authors">7. Authors</h2>
<ul>
<li><p><a href="https://github.com/anitalasa">Ana V. Lasa</a> <a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Antonio José Fernández-González <a href="https://orcid.org/0000-0002-1938-5436" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Pablo J. Villadas <a href="https://orcid.org/0000-0001-8662-8883" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Carmen Gómez-Lama Cabanás <a href="https://orcid.org/0000-0001-8312-9253" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Antonio Valverde Corredor <a href="https://orcid.org/0000-0001-5983-2045" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Manuel Fernández-López <a href="https://orcid.org/0000-0003-2366-6908" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
<li><p>Jesús Mercado-Blanco <a href="https://orcid.org/0000-0003-1895-5895" target="orcid.widget"> <img src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" alt="ORCID logo" width="16" height="16"></a></p></li>
</ul>
</section>
<section id="how-to-cite" class="level2">
<h2 class="anchored" data-anchor-id="how-to-cite">8. How to cite</h2>
<section id="main-paper" class="level3">
<h3 class="anchored" data-anchor-id="main-paper">8.1 Main paper</h3>
<p>Still building</p>
</section>
<section id="repository" class="level3">
<h3 class="anchored" data-anchor-id="repository">8.2 Repository</h3>
<p>The repository is available at this <a href="https://anitalasa.github.io/BananaMicrobiota/" title="Click to the repository">website</a> (https://anitalasa.github.io/BananaMicrobiota/)</p>
<p>You can also find <a href="https://anitalasabanana.netlify.app" title="Click to the repository">here</a> (https://anitalasabanana.netlify.app/)</p>
</section>
<section id="references" class="level3">
<h3 class="anchored" data-anchor-id="references">9. References</h3>
<p>Ihrmark, K., Bödeker, I.T.M., Cruz-Martinez, K., Friberg, H., Kubartova, A., et al., 2012. New primers to amplify the fungal ITS2 region – evaluation by 454- sequencing of artificial and natural communities. FEMS Microbiol. Ecol. 82, 666–677. https://doi.org/10.1111/j.1574-6941.2012.01437.x</p>
<p>Lundberg, D.S., Yourstone, S., Mieczkowski, P., Jones, C.D., Dangl, J.L., 2013. Practical innovations for high-throughput amplicon sequencing. Nat. Methods. 10, 999–1002. https://doi.org/10.1038/nmeth.2634</p>
<p>Takahashi, S., Tomita, J., Nishioka, K., Hisada, T., Nishijima, M., 2014. Development of a prokaryotic universal primer for simultaneous analysis of bacteria and archaea using next-generation sequencing. PloS One. 9, 8. https://doi.org/10.1371/journal. pone.0105592.</p>
<p>White, T.J., Bruns, T.D., Lee, S.B., Taylor, J.W., 1990. Amplification and direct sequencing of fungal ribosomal RNA genes for phylogenetics, Iin: Innis, M.A., Gelfand, D.H., Sninsky, J.J., White, T.J. (Eds.), PCR Protocols: A Guide to Methods and Applications. Academics Press, San Diego, pp.&nbsp;315–322.</p>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "The microbiota of banana plants"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">## Scripts to process and analyze high-throughput sequencing reads</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="al">![Figure 1. Graphic representation of banana plants](images/Dibujo.jpg)</span>{fig-align="center" width="500"}</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>This repository includes all the scripts employed for the processing of amplicon sequencing reads and the scripts needed for all the subsequent ecological analyses.</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">## 1. Introduction</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>These analyses were made for the project **EVOMICROBIA** (*Evolución dirigida de consorcios microbianos mejorados para el biocontrol de la Fusariosis vascular del Plátano de Canarias*). The main aim of this work was to analyze the microbial communities (bacteria and fungi) associated to the roots of banana plants (*Musa acuminata* cv. Pequeña Enana synonymous with Dwarf Cavendish). We also aimed at determining whether microbial members are transferred from mother to sucker plants through the root system, and if possible, to decipher which part of the microbiota is transferred from mothers to first suckers and from these to second sucker plants. We focused on two plots(orchards) located in different points in Tenerife Island (Canary Islands, Spain), with different agronomic management. In the following map, the location of both plots its indicated (Figure 2):</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="al">![Figure 2. Map of Tenerife Island (Canary Islands, Spain) and photographs of both orchards under study](images/Mapa.jpg)</span>{fig-align="center" width="500"}</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">### 1.1 But.. how do banana plants develop?</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>Banana mats are comprised of the underground rhizome (also named *corm*) from which “suckers” or new clonal shoots arise close to mother plants, the pseudostem (composed by tightly packed overlapping leaf sheaths), stem and leafs. Thus, suckers are forms of vegetative reproduction, and can develop and produce an inflorescence when they reach the adult state. Once mother plants produce fruit bunches and they are harvested, mother plants are cut since they do not produce bananas any more. This cycle repeats periodically, so suckers become mothers and so happens with all the clonal progeny (clonal "offspring").</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">## 2. Experimental design</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>Samples were taken in two consecutive campaigns:</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="ss">1.  </span>**Year 2022**: 12 healthy banana mother plants whose sucker plants were actively growing were selected from each of the orchards. Samples from rhizosphere soil of the 12 mother plants were manually collected, along with rhizosphere soils from the sucker plant growing close to the each mother. Roots from each mother and sucker plants were also collected in each orchard Thus, a total of 12 replicates x 2 plants in different developmental state (mother and sucker) x 2 orchards x 2 plant compartments = 96 samples.</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="ss">2.  </span>**Year 2023**: the same sampling scheme was followed. However, it should be taken into account that plants that were suckers in 2022 acted as mothers in 2023, and their suckers were considered as the second suckers.</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>Have a look at the next table that summarizes the sampling scheme.</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>|               |                 |                    |            |                    |            |</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>|:----------:|:----------:|:----------:|:----------:|:----------:|:----------:|</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>| **Year 2022** |                 |                    |            |                    |            |</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>|               |                 | **Plot 1 (South)** |            | **Plot 2 (North)** |            |</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>|               |                 |     **Mother**     | **Sucker** |     **Mother**     | **Sucker** |</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>|               | **Rhizosphere** |         12         |     12     |         12         |     12     |</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>|               | **Endosphere**  |         12         |     12     |         12         |     12     |</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>| **Year 2023** |                 |                    |            |                    |            |</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>|               | **Rhizosphere** |         12         |     12     |         12         |     12     |</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>|               | **Endosphere**  |         12         |     12     |         12         |     12     |</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>: *Number of samples analyzed in each year, plant compartment and plot*.</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="fu">## 3. Sequencing scheme</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>We started from 192 samples, from which total DNA was isolated (in the case of the roots, they were surface sterilized previously). Thus, we got 192 DNA samples. However, it should be taken into account that we wanted to analyze both bacterial and fungal communities. Thus, we had to sequence 192 x 2 = 384 samples (and also the controls). Due to the high amount of samples and the different sampling campaigns, our sequencing plan was divided into different runs. Furthermore, the sequencing service (<span class="co">[</span><span class="ot">*Instituto de Parasitología y Biomedicina López-Neyra*</span><span class="co">](https://www.ipb.csic.es/servicios/Genomica/index.html "Click to Genomics Service")</span>, CSIC, Granada) had to resequence some samples due to technical issues. Thus, our sequencing project had to be divided into the following sequencing runs:</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Bacteria:</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 1: most of the samples corresponding to the first sampling campaign (2022).</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 2: some additional samples corresponding to year 2022.</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 3: samples corresponding to the second campaign (2023).</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 4: resequencing of some samples corresponding to year 2023.</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- --&gt;</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Fungi:</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 5: most of the samples corresponding to the first campaign (2022).</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 6: other samples corresponding to 2022.</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="ss">    -   </span>Run 7: all the samples corresponding to 2023</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>It should be pointed out that in each sequencing run (just in the case of the bacterial dataset) three replicates of the mock community ZymoBIOMICS Microbial Community Standard II (ZYMO Research) were included.</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="fu">## 4. Sample naming</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>The name of the samples was composed by a combination of alphanumeric characters.</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>First character: letter "A" for samples obtained in 2022, and "B" for those obtained in 2023.</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Second character: numer "1" for the plot located in the south orchard, and "2" for the plot in the north of the island.</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Third character: letter "M" for mother plants, and "S" for suckers.</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Fourth and fifth character: the number of the replicate. It should be taken into account that from 1 to 9, replicates are names as 01-09.</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Sixth character: letter "E" for root endosphere and "R" for rhizosphere soil samples.</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>Thus, rhizosphere soil of the mother plant number 7 located in the north orchard sampled in 2023 will be named as B2M07E.</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="fu">## 5. Technical issues</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>-In the case of **bacteria**, the hypervariable regions V3-V4 of the prokaryotic gene *16S rRNA* were sequenced. For that purpose, the primer pair Pro341F and Pro805R was employed (<span class="co">[</span><span class="ot">Takahashi et al., 2014</span><span class="co">](https://doi.org/10.1371/journal.pone.0105592 "Go to the paper")</span>)</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>-In the case of **fungi**, the ITS2 region was sequenced. For that purpose, the primer pair ITS4 and fiTS7 was employed (White et al., 1990 and <span class="co">[</span><span class="ot">Ihrmark et al., 2012</span><span class="co">](https://doi.org/10.1111/j.1574-6941.2012.01437.x "Go to the paper")</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>-In order to diminish the amplification and sequencing of the DNA from the plant host (from mitochondria, chloroplast etc), DNA was treated with the PNA PCR Clamps proposed by <span class="co">[</span><span class="ot">Lunderberg et al., (2013)</span><span class="co">](https://doi.org/10.1038/nmeth.2634 "Go to the paper")</span></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>-MiSeq Illumina platform was used for the sequencing, and two types of Paired-End strategies were followed:</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Runs 1, 2, 3, 5, 6 and 7**: 2x275 PE strategy</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**Run 4**: 2x300 PE strategy</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>The remaining technical issues are explained in the Research Article</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="fu">## 6. What will I find in this GitHub project?</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>In this repository, you will find all the scripts that we used for two types of approaches: 1) Processing of Illumina MiSeq reads and 2) Ecological analyses</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a><span class="fu">### 6.1 Scripts for the processing of MiSeq reads</span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>These scripts explain all the steps to process the Illumina reads, from the fastq files to a filtered ASV table.</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">16S rRNA reads</span><span class="co">](Bacteria_reads_processing.qmd)</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">ITS2 reads</span><span class="co">](Fungi_reads_processing.qmd)</span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="fu">### 6.2 Script for the ecological analyses</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>All the steps from the previously obtained ASV table to the alpha and beta diversity, differential abundance analysis, etc.</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Analysis of bacterial dataset</span><span class="co">](Bacteria_data_analysis.qmd)</span></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Analysis of fungal dataset</span><span class="co">](Fungi_data_analysis.qmd)</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="fu">## 7. Authors</span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="co">[</span><span class="ot">Ana V. Lasa</span><span class="co">](https://github.com/anitalasa)</span> <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0003-3783-7157"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Antonio José Fernández-González <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0002-1938-5436"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Pablo J. Villadas <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0001-8662-8883"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Carmen Gómez-Lama Cabanás <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0001-8312-9253"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Antonio Valverde Corredor <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0001-5983-2045"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Manuel Fernández-López <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0003-2366-6908"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Jesús Mercado-Blanco <span class="kw">&lt;a</span> <span class="er">href</span><span class="ot">=</span><span class="st">"https://orcid.org/0000-0003-1895-5895"</span> <span class="er">target</span><span class="ot">=</span><span class="st">"orcid.widget"</span><span class="kw">&gt;</span> <span class="kw">&lt;img</span> <span class="er">src</span><span class="ot">=</span><span class="st">"https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png"</span> <span class="er">alt</span><span class="ot">=</span><span class="st">"ORCID logo"</span> <span class="er">width</span><span class="ot">=</span><span class="st">"16"</span> <span class="er">height</span><span class="ot">=</span><span class="st">"16"</span><span class="kw">/&gt;&lt;/a&gt;</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a><span class="fu">## 8. How to cite</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="fu">### 8.1 Main paper</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>Still building</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a><span class="fu">### 8.2 Repository</span></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>The repository is available at this <span class="co">[</span><span class="ot">website</span><span class="co">](https://anitalasa.github.io/BananaMicrobiota/ "Click to the repository")</span> (https://anitalasa.github.io/BananaMicrobiota/)</span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>You can also find <span class="co">[</span><span class="ot">here</span><span class="co">](https://anitalasabanana.netlify.app "Click to the repository")</span> (https://anitalasabanana.netlify.app/)</span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a><span class="fu">### 9. References</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>Ihrmark, K., Bödeker, I.T.M., Cruz-Martinez, K., Friberg, H., Kubartova, A., et al., 2012. New primers to amplify the fungal ITS2 region – evaluation by 454- sequencing of artificial and natural communities. FEMS Microbiol. Ecol. 82, 666–677. https://doi.org/10.1111/j.1574-6941.2012.01437.x</span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a>Lundberg, D.S., Yourstone, S., Mieczkowski, P., Jones, C.D., Dangl, J.L., 2013. Practical innovations for high-throughput amplicon sequencing. Nat. Methods. 10, 999–1002. https://doi.org/10.1038/nmeth.2634</span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a>Takahashi, S., Tomita, J., Nishioka, K., Hisada, T., Nishijima, M., 2014. Development of a prokaryotic universal primer for simultaneous analysis of bacteria and archaea using next-generation sequencing. PloS One. 9, 8. https://doi.org/10.1371/journal. pone.0105592.</span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>White, T.J., Bruns, T.D., Lee, S.B., Taylor, J.W., 1990. Amplification and direct sequencing of fungal ribosomal RNA genes for phylogenetics, Iin: Innis, M.A., Gelfand, D.H., Sninsky, J.J., White, T.J. (Eds.), PCR Protocols: A Guide to Methods and Applications. Academics Press, San Diego, pp. 315–322.</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>